{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import os\n",
    "import math\n",
    "import bisect\n",
    "import altair as alt\n",
    "pd.options.mode.chained_assignment = None\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1.8606258759661092"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Scaling factor for entrapment calculations, based on the ratio on the number of entrapment peptide sequences and human peptide sequences in the database\n",
    "\n",
    "scaling_factor = 1 + 3105275 / 3608159\n",
    "scaling_factor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "entrapment_seq_path = r\"C:\\Users\\Alex\\Documents\\Proteomes\\MBR_Proteomes_March_2024\\EntrapmentProteinPeptideSequences_50percent.txt\"\n",
    "with open(entrapment_seq_path) as f:\n",
    "    entrapment_seqs = f.readlines()\n",
    "\n",
    "entrapment_seqs_set = set()\n",
    "for seq in entrapment_seqs:\n",
    "    entrapment_seqs_set.add(seq.replace(\"\\n\", \"\"))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Build a dictionary linking protein accessions to their species of origin (For MaxQuant analysis)\n",
    "accession_dict = dict()\n",
    "\n",
    "yeast_fasta_sequences = SeqIO.parse(open(r\"C:\\Users\\Alex\\Documents\\Proteomes\\MBR_Proteomes_March_2024\\uniprot_SCerevisiae_6k_03_2024.fasta\"),'fasta')\n",
    "yeast_records = list(yeast_fasta_sequences)\n",
    "for record in yeast_records:\n",
    "    accession_dict[record.id.split('|')[1]] = 'Yeast'\n",
    "\n",
    "human_fasta_sequences = SeqIO.parse(open(r\"C:\\Users\\Alex\\Documents\\Proteomes\\MBR_Proteomes_March_2024\\uniprot_HSapiens_80k_03_2024.fasta\"),'fasta')\n",
    "human_records = list(human_fasta_sequences)\n",
    "for record in human_records:\n",
    "    accession_dict[record.id.split('|')[1]] = 'Human'\n",
    "\n",
    "human_fasta_sequences = SeqIO.parse(open(r\"C:\\Users\\Alex\\Documents\\Proteomes\\MBR_Proteomes_March_2024\\ConcatenatedHumanEntrapmentProteins_50percent.fasta\"),'fasta')\n",
    "human_records = list(human_fasta_sequences)\n",
    "for record in human_records:\n",
    "    accession_dict[record.id.split('|')[1]] = 'Human'\n",
    "    accession_dict[record.id] = 'Human'\n",
    "\n",
    "arabida_fasta_sequences = SeqIO.parse(open(r\"C:\\Users\\Alex\\Documents\\Proteomes\\MBR_Proteomes_March_2024\\uniprot_Arabidopsis_40k_03_2024.fasta\"),'fasta')\n",
    "arabida_records = list(arabida_fasta_sequences)\n",
    "for record in arabida_records:\n",
    "    accession_dict[record.id.split('|')[1]] = 'Arabida'\n",
    "\n",
    "entrapment_fasta_sequences = SeqIO.parse(open(r\"C:\\Users\\Alex\\Documents\\Proteomes\\MBR_Proteomes_March_2024\\HumanEntrapmentDb.fasta\"), 'fasta')\n",
    "entrapment_records = list(entrapment_fasta_sequences)\n",
    "for record in entrapment_records:\n",
    "    accession_dict[record.id.split('|')[1]] = 'Arabida'\n",
    "    \n",
    "entrapment_formatted_seq = SeqIO.parse(open(r\"\\\\bison.chem.wisc.edu\\share\\Users\\AlexanderS_Bison\\MBR Data\\Proteomes\\HumanEntrapmentDb_Formatted.fasta\"), 'fasta')\n",
    "entrapment_records = list(entrapment_fasta_sequences)\n",
    "for record in entrapment_records:\n",
    "    accession_dict[record.id.split('|')[1]] = 'Arabida'\n",
    "    accession_dict[record] = 'Arabida'\n",
    "\n",
    "ecoli_fasta_sequences = SeqIO.parse(open(r\"C:\\Users\\Alex\\Documents\\Proteomes\\MBR_Proteomes_March_2024\\uniprot_Ecoli_4k_03_2024.fasta\"),'fasta')\n",
    "ecoli_records = list(ecoli_fasta_sequences)\n",
    "for record in ecoli_records:\n",
    "    accession_dict[record.id.split('|')[1]] = 'Ecoli'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_fdp_old_flash(o_peaks_path, c_peaks_path, censored_psm_path, human_file_pattern = \"_1x02nguL_\",\n",
    "                   foreign_species = 'Yeast', rt_delta = 0.25):\n",
    "\n",
    "    # calculate the error rate for (presumably) native human transfers\n",
    "    peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
    "    \n",
    "    peaks = peaks.loc[~peaks[\"Protein Group\"].str.contains(\"DECOY\")]\n",
    "    peaks = peaks.loc[peaks[\"File Name\"].str.contains(human_file_pattern)]\n",
    "    peaks[\"Organism\"] = peaks.apply(get_organisms, axis = 1)\n",
    "\n",
    "    msms_detected_yeast_peaks = peaks.loc[(peaks[\"Peak Detection Type\"] == \"MSMS\") &\n",
    "                           (peaks[\"Organism\"].str.contains(foreign_species))]\n",
    "    msms_detected_yeast_peaks.drop_duplicates(subset = [\"Full Sequence\"], keep = 'first', inplace = True)\n",
    "    msms_yeast_seqs = set(msms_detected_yeast_peaks[\"Full Sequence\"].tolist())\n",
    "\n",
    "    peaks = peaks.loc[(peaks[\"Peak Detection Type\"] == \"MBR\")]\n",
    "\n",
    "    human_peaks = peaks.loc[peaks[\"Organism\"].str.contains('Human')]\n",
    "    # Our entrapment database has a bunch of scrambled sequences appended to the end of the DB, which is why we're checking for them\n",
    "    entrapment_peaks = human_peaks.loc[human_peaks[\"Full Sequence\"].isin(entrapment_seqs_set)]\n",
    "    human_peaks = human_peaks.loc[~human_peaks[\"Base Sequence\"].isin(entrapment_seqs_set)]\n",
    "    \n",
    "    human_count = human_peaks.shape[0]\n",
    "    arabida_count = entrapment_peaks.shape[0]\n",
    "\n",
    "    yeast_peaks = peaks.loc[peaks[\"Organism\"].str.contains(foreign_species)]\n",
    "    yeast_peaks = yeast_peaks.loc[~yeast_peaks[\"Organism\"].str.contains('Human')]\n",
    "    yeast_count = yeast_peaks.shape[0]\n",
    "    \n",
    "    msms_yeast_count = sum(yeast_peaks[\"Full Sequence\"].isin(msms_yeast_seqs))\n",
    "    yeast_count = yeast_count - msms_yeast_count\n",
    "\n",
    "    #arabida_peaks = peaks.loc[peaks[\"Organism\"].str.contains(\"Arabida\")]\n",
    "    #arabida_count = arabida_peaks.shape[0]\n",
    "\n",
    "    scaled_arabida = arabida_count * scaling_factor\n",
    "    nper_results = get_nper_old_flash(censored_psm_path, o_peaks_path, c_peaks_path, rt_delta)\n",
    "    scaled_nper = human_count * nper_results[\"error_rate\"]\n",
    "\n",
    "    total = human_count + yeast_count + arabida_count\n",
    "\n",
    "    fper = 100 * yeast_count / total\n",
    "    nper = 100 * scaled_nper / total\n",
    "    fder = (100 - fper) * scaled_arabida / total\n",
    "\n",
    "    false_discovery_proportion = fper + nper + fder\n",
    "\n",
    "    results = dict(\n",
    "        {\n",
    "            \"human\": human_count,\n",
    "            \"foreign\": yeast_count,\n",
    "            \"arabida\": arabida_count,\n",
    "            \"total\": total,\n",
    "            \"scaled_arabida\": scaled_arabida,\n",
    "            \"scaled_nper\": scaled_nper,\n",
    "            \"sensitivity\": nper_results[\"sensitivity\"],\n",
    "            \"fper\": fper,\n",
    "            \"nper\": nper,\n",
    "            \"fder\": fder,\n",
    "            \"false_discovery_proportion\": false_discovery_proportion\n",
    "        }\n",
    "    )\n",
    "    return results\n",
    "\n",
    "def get_organisms(row):\n",
    "    accesions = row[\"Protein Group\"].split(\";\")\n",
    "    organisms = []\n",
    "    for a in accesions:\n",
    "        if a in accession_dict:\n",
    "            organisms.append(accession_dict[a])\n",
    "        elif (\"ENTRAPMENT\" in a):\n",
    "            organisms.append(\"Arabida\")\n",
    "    return \";\".join(organisms)\n",
    "\n",
    "def get_nper_old_flash(censored_path, old_peak_path, new_peak_path, rt_delta = 0.25):\n",
    "    # Read in the list of peptides that were censored\n",
    "    censored_psms = pd.read_csv(censored_path, sep = '\\t')\n",
    "    # Select the ones that were quantified in the initial FlashLFQ analysis\n",
    "    original_peaks = pd.read_csv(old_peak_path, sep = '\\t')\n",
    "    original_peaks = original_peaks[original_peaks[\"Peak RT Apex\"] != \"-\"]\n",
    "    censored_peaks = pd.merge(censored_psms, original_peaks, how = \"inner\", left_on=[\"File Name\", \"Full Sequence\"], right_on=[\"File Name\", \"Full Sequence\"])\n",
    "    censored_peaks = censored_peaks[[\"File Name\", \"Full Sequence\", \"Peak RT Start\", \"Peak RT Apex\", \"Peak RT End\", \"Peak Charge\", \"Peak intensity\"]]\n",
    "    # This will allow for comparison later, as all the new peaks were take from files named in this way\n",
    "    censored_peaks[\"File Name\"] = censored_peaks[\"File Name\"].astype(str) + \"-censored\" \n",
    "\n",
    "    # Merge the old and new results\n",
    "    new_peaks = pd.read_csv(new_peak_path, sep = '\\t')\n",
    "    new_peaks = new_peaks.loc[(new_peaks[\"Peak Detection Type\"] == \"MBR\")]\n",
    "    new_peaks = new_peaks[[\"File Name\", \"Full Sequence\", \"Peak RT Start\", \"Peak RT Apex\", \"Peak RT End\", \"Peak Charge\",  \"Peak intensity\", \"MBR Score\"]]\n",
    "    peak_join = pd.merge(censored_peaks, new_peaks, how = \"inner\", left_on=[\"File Name\", \"Full Sequence\"], right_on=[\"File Name\", \"Full Sequence\"])\n",
    "\n",
    "    peak_join_no_dup = peak_join.drop_duplicates(subset = [\"File Name\", \"Full Sequence\"], keep = 'first')\n",
    "    censored_peaks_no_dup = censored_peaks.drop_duplicates(subset = [\"File Name\", \"Full Sequence\"], keep = 'first')\n",
    "    sensitivity = peak_join_no_dup.shape[0] / censored_peaks_no_dup.shape[0]\n",
    "\n",
    "    #compare the old and new results (here, 1 means the match, 0 means they dont)\n",
    "    peak_join[\"Agreement\"] = peak_join.apply(lambda x: check_overlap_time(x, rt_delta), axis = 1)\n",
    "\n",
    "    if(peak_join.shape[0] == 0):\n",
    "        return dict({\n",
    "            \"sensitivity\": 0,\n",
    "            \"error_rate\": 0} )\n",
    "\n",
    "    peak_join.sort_values(by = 'MBR Score', ascending=False, inplace=True)\n",
    "    peak_join[\"good_transfers\"] = (peak_join[\"Agreement\"] == 1).cumsum()\n",
    "    peak_join[\"bad_transfers\"] = (peak_join[\"Agreement\"] == 0).cumsum()\n",
    "    peak_join[\"error_rate\"] = peak_join.apply(calculateErrorRate, axis = 1)\n",
    "\n",
    "    results = dict(\n",
    "        {\n",
    "            \"sensitivity\": sensitivity,\n",
    "            \"error_rate\": peak_join[\"error_rate\"].iloc[-1]\n",
    "        }\n",
    "\n",
    "    )\n",
    "\n",
    "    return results\n",
    "\n",
    "# Native Peak Error Rate functions\n",
    "def check_overlap_time(table, rt_delta = 0.25):\n",
    "    try:\n",
    "        pip_apex = float(table[\"Peak RT Apex_y\"])\n",
    "        ms_apex = float(table[\"Peak RT Apex_x\"])\n",
    "    except:\n",
    "        return -1\n",
    "    if(abs(pip_apex-ms_apex) < rt_delta):\n",
    "        return 1\n",
    "    else:\n",
    "        return 0\n",
    "    \n",
    "    \n",
    "def calculateErrorRate(table):\n",
    "    return table[\"bad_transfers\"] / (table[\"good_transfers\"] + table[\"bad_transfers\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_fdp_flash(o_peaks_path, c_peaks_path, censored_psm_path, human_file_pattern = \"_1x02nguL_\",\n",
    "                   foreign_species = 'Saccharomyces cerevisiae', q_value = 0.05, rt_delta = 0.5):\n",
    "\n",
    "    peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
    "    peaks = peaks.loc[peaks[\"File Name\"].str.contains(human_file_pattern)]\n",
    "\n",
    "    msms_detected_yeast_peaks = peaks.loc[(peaks[\"Peak Detection Type\"] == \"MSMS\") &\n",
    "                           (peaks[\"Organism\"].str.contains(foreign_species))]\n",
    "    msms_detected_yeast_peaks.drop_duplicates(subset = [\"Full Sequence\"], keep = 'first', inplace = True)\n",
    "    msms_yeast_seqs = msms_detected_yeast_peaks[\"Full Sequence\"].tolist()\n",
    "\n",
    "    msms_peaks = peaks.loc[(peaks[\"Peak Detection Type\"] == \"MSMS\") &\n",
    "                           (peaks[\"Decoy Peptide\"] == False) &\n",
    "                           ((peaks[\"Organism\"].str.contains(foreign_species)) | (peaks[\"Organism\"].str.contains('Homo sapiens')) | (peaks[\"Organism\"].str.contains('Arabidopsis')))]\n",
    "    msms_count = msms_peaks.shape[0]\n",
    "\n",
    "    peaks = peaks.loc[(peaks[\"Peak Detection Type\"] == \"MBR\") & (peaks[\"PIP Q-Value\"] < q_value) & (peaks[\"Decoy Peptide\"] == False)]\n",
    "\n",
    "    peaks = peaks.loc[(peaks[\"Random RT\"] == False) & (peaks[\"Decoy Peptide\"] == False)]\n",
    "\n",
    "    human_peaks = peaks.loc[peaks[\"Organism\"].str.contains('Homo sapiens')]\n",
    "    # Our entrapment database has a bunch of scrambled sequences appended to the end of the DB, which is why we're checking for them\n",
    "    entrapment_peaks = human_peaks.loc[human_peaks[\"Full Sequence\"].isin(entrapment_seqs_set)]\n",
    "    human_peaks = human_peaks.loc[~human_peaks[\"Base Sequence\"].isin(entrapment_seqs_set)]\n",
    "    \n",
    "    human_count = human_peaks.shape[0]\n",
    "    arabida_count = entrapment_peaks.shape[0]\n",
    "\n",
    "    yeast_peaks = peaks.loc[peaks[\"Organism\"].str.contains(foreign_species)]\n",
    "    yeast_peaks = yeast_peaks.loc[~yeast_peaks[\"Organism\"].str.contains('Homo sapiens')]\n",
    "    yeast_count = yeast_peaks.shape[0]\n",
    "    \n",
    "    msms_yeast_count = sum(yeast_peaks[\"Full Sequence\"].isin(msms_yeast_seqs))\n",
    "    yeast_count = yeast_count - msms_yeast_count\n",
    "\n",
    "    #arabida_peaks = peaks.loc[peaks[\"Organism\"].str.contains(\"ENTRAPMENT\")]\n",
    "    #arabida_peaks = arabida_peaks.loc[~arabida_peaks[\"Organism\"].str.contains(foreign_species)]\n",
    "    #arabida_count = arabida_peaks.shape[0]\n",
    "\n",
    "    score_threshold = peaks[\"PIP PEP\"].max()\n",
    "\n",
    "    # calculate the error rate for (presumably) native human transfers\n",
    "    nper_results = get_native_peak_error_rate( \n",
    "                            o_peaks_path = o_peaks_path,\n",
    "                            c_peaks_path = c_peaks_path,\n",
    "                            censored_psm_path = censored_psm_path,\n",
    "                            q_value_threshold = q_value,\n",
    "                            score_threshold = score_threshold,\n",
    "                            rt_delta = rt_delta)\n",
    "    \n",
    "\n",
    "    scaled_arabida = arabida_count * scaling_factor\n",
    "    scaled_nper = human_count * nper_results[\"error_rate\"]\n",
    "\n",
    "    total = human_count + yeast_count + arabida_count\n",
    "    total = max(1, total)\n",
    "\n",
    "    fper = 100 * yeast_count / total\n",
    "    nper = 100 * scaled_nper / total\n",
    "    fder = (100-fper) * (scaled_arabida / total)\n",
    "\n",
    "    false_discovery_proportion = fper + nper + fder\n",
    "\n",
    "    results = dict(\n",
    "        {\n",
    "            \"human\": human_count,\n",
    "            \"foreign\": yeast_count,\n",
    "            \"arabida\": arabida_count,\n",
    "            \"total\": total,\n",
    "            \"msms_count\" : msms_count,\n",
    "            \"scaled_arabida\": scaled_arabida,\n",
    "            \"scaled_nper\": scaled_nper,\n",
    "            \"sensitivity\": nper_results[\"sensitivity\"],\n",
    "            \"fper\": fper,\n",
    "            \"nper\": nper,\n",
    "            \"fder\": fder,\n",
    "            \"false_discovery_proportion\": false_discovery_proportion\n",
    "        }\n",
    "    )\n",
    "    return results\n",
    "\n",
    "# Native Peak Error Rate functions\n",
    "def check_overlap_time(table, rt_delta = 0.25):\n",
    "    try:\n",
    "        pip_apex = float(table[\"Peak RT Apex_y\"])\n",
    "        ms_apex = float(table[\"Peak RT Apex_x\"])\n",
    "    except:\n",
    "        return -1\n",
    "    if(abs(pip_apex-ms_apex) < rt_delta):\n",
    "        return 1\n",
    "    else:\n",
    "        return 0\n",
    "    \n",
    "def calculateErrorRate(table):\n",
    "    return table[\"bad_transfers\"] / (table[\"good_transfers\"] + table[\"bad_transfers\"])\n",
    "\n",
    "def get_native_peak_error_rate(o_peaks_path, c_peaks_path, censored_psm_path, q_value_threshold, score_threshold=10, rt_delta = 0.25):\n",
    "    # Read in the list of peptides that were censored\n",
    "    censored_psms = pd.read_csv(censored_psm_path, sep = '\\t')\n",
    "    # Select the ones that were quantified in the initial FlashLFQ analysis\n",
    "    original_peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
    "    original_peaks = original_peaks[original_peaks[\"Peak RT Apex\"] != \"-\"]\n",
    "    \n",
    "    censored_peaks = pd.merge(censored_psms, original_peaks, how = \"inner\", left_on=[\"File Name\", \"Full Sequence\"], right_on=[\"File Name\", \"Full Sequence\"])\n",
    "    censored_peaks = censored_peaks[[\"File Name\", \"Full Sequence\", \"Peak RT Start\", \"Peak RT Apex\", \"Peak RT End\", \"Peak Charge\", \"Peak intensity\"]]\n",
    "\n",
    "    # This will allow for comparison later, as all the new peaks were take from files named in this way\n",
    "    censored_peaks[\"File Name\"] = censored_peaks[\"File Name\"].astype(str) + \"-censored\" \n",
    "\n",
    "    # Merge the old and new results\n",
    "    new_peaks = pd.read_csv(c_peaks_path, sep = '\\t')\n",
    "    new_peaks = new_peaks.loc[(new_peaks[\"Peak Detection Type\"] == \"MBR\") & (new_peaks[\"Random RT\"] == False) & (new_peaks[\"Decoy Peptide\"] == False)]\n",
    "    new_peaks = new_peaks.loc[(new_peaks[\"PIP Q-Value\"] < q_value_threshold)]\n",
    "    #new_peaks = new_peaks.loc[(new_peaks[\"MBR Score\"] > score_threshold)]\n",
    "    new_peaks = new_peaks[[\"File Name\", \"Full Sequence\", \"Peak RT Start\", \"Peak RT Apex\", \"Peak RT End\", \"Peak Charge\",  \"Peak intensity\", \"PIP PEP\", 'PIP Q-Value']]\n",
    "    peak_join = pd.merge(censored_peaks, new_peaks, how = \"inner\", left_on=[\"File Name\", \"Full Sequence\"], right_on=[\"File Name\", \"Full Sequence\"])\n",
    "\n",
    "    peak_join[\"Agreement\"] = peak_join.apply(lambda x: check_overlap_time(x, rt_delta), axis = 1)\n",
    "    peak_join.sort_values(by = \"Agreement\", ascending=False, inplace=True)\n",
    "\n",
    "    #compare the old and new results (here, 1 means the match, 0 means they dont)\n",
    "    peak_join.drop_duplicates(subset = [\"File Name\", \"Full Sequence\"], keep = 'first', inplace=True)\n",
    "    censored_peaks.drop_duplicates(subset = [\"File Name\", \"Full Sequence\"], keep = 'first', inplace=True)\n",
    "    sensitivity = peak_join.shape[0] / censored_peaks.shape[0]\n",
    "\n",
    "\n",
    "    if(peak_join.shape[0] == 0):\n",
    "        return dict({\n",
    "            \"sensitivity\": 0,\n",
    "            \"error_rate\": 0} )\n",
    "\n",
    "    peak_join.sort_values(by = 'PIP Q-Value', ascending=True, inplace=True)\n",
    "    peak_join[\"good_transfers\"] = (peak_join[\"Agreement\"] == 1).cumsum()\n",
    "    peak_join[\"bad_transfers\"] = (peak_join[\"Agreement\"] == 0).cumsum()\n",
    "    peak_join[\"error_rate\"] = peak_join.apply(calculateErrorRate, axis = 1)\n",
    "\n",
    "    results = dict(\n",
    "        {\n",
    "            \"sensitivity\": sensitivity,\n",
    "            \"error_rate\": peak_join[\"error_rate\"].iloc[-1]\n",
    "        }\n",
    "    )\n",
    "\n",
    "    return results\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_fdp_fragger(o_folder = r\"D:\\GygiTwoProteome_PXD014415\\IonQuant1Percent\",\n",
    "                      c_folder= r\"D:\\GygiTwoProteome_PXD014415\\IonQuant1Percent_censored\",\n",
    "                      censored_psm_path = r\"D:\\GygiTwoProteome_PXD014415\\CensoredDataFiles_fragger\\CensoredPsms.tsv\",\n",
    "                      foreign_species = \"YEAST\",\n",
    "                      human_file_pattern = \"_human_90min_\",\n",
    "                      rt_diff_threshold = 0.5):\n",
    "        \n",
    "        rt_diff_threshold = rt_diff_threshold * 60 # Convert to seconds, which is how iq stores RT\n",
    "\n",
    "        human_rep_indices = get_indices(o_folder, human_file_pattern)\n",
    "        human_rep_indices_censored = get_indices(c_folder, human_file_pattern)\n",
    "        \n",
    "        ion_count = count_ions_by_species(o_folder + r\"\\combined_modified_peptide.tsv\", match_indices = human_rep_indices, foreign_species = foreign_species)\n",
    "        nper_results = analyze_nper(o_folder = o_folder, c_folder = c_folder, censored_psm_path = censored_psm_path, \n",
    "                                         human_rep_indices = human_rep_indices, human_rep_indices_censored = human_rep_indices_censored,\n",
    "                                         rt_diff_threshold = rt_diff_threshold)\n",
    "        \n",
    "        scaled_arabida = ion_count[\"arabida\"] * scaling_factor\n",
    "        scaled_nper = ion_count[\"human\"] * nper_results[\"specificity\"]\n",
    "\n",
    "        ion_count[\"scaled_arabida\"] = scaled_arabida\n",
    "        ion_count[\"scaled_nper\"] = scaled_nper\n",
    "        ion_count[\"specificity\"] = nper_results[\"specificity\"]\n",
    "        ion_count[\"sensitivity\"] = nper_results[\"sensitivity\"]\n",
    "        ion_count[\"nper_pct_diff\"] = nper_results[\"intensity_diff\"]\n",
    "\n",
    "        ion_count[\"fper\"] = 100 * ion_count[\"foreign\"]  / ion_count[\"total\"]\n",
    "        ion_count[\"nper\"] = 100 * ion_count[\"scaled_nper\"]  / ion_count[\"total\"]\n",
    "        ion_count[\"fder\"] = (100-ion_count[\"fper\"]) * ion_count[\"scaled_arabida\"]  / ion_count[\"total\"]\n",
    "        ion_count[\"false_discovery_proportion\"] = ion_count[\"fper\"] + ion_count[\"nper\"] + ion_count[\"fder\"]\n",
    "\n",
    "        return ion_count\n",
    "    \n",
    "def get_indices(folder_path, pattern = \"_human_90min_\"):\n",
    "    exp_file_path = folder_path + r\"\\experiment_annotation.tsv\"\n",
    "    exp_file = pd.read_csv(exp_file_path, sep = '\\t')\n",
    "    human_indices = exp_file[exp_file[\"file\"].str.contains(pattern)].index\n",
    "    return human_indices\n",
    "\n",
    "def analyze_nper(o_folder,\n",
    "                      c_folder,\n",
    "                      censored_psm_path,\n",
    "                      human_rep_indices,\n",
    "                      human_rep_indices_censored = None,\n",
    "                      rt_diff_threshold = 30):\n",
    "    \n",
    "    if(not any(human_rep_indices)):\n",
    "        human_rep_indices_censored = human_rep_indices\n",
    "\n",
    "    o_annotation_path = o_folder + r\"\\experiment_annotation.tsv\"\n",
    "    annotation_file = pd.read_csv(o_annotation_path, sep = '\\t')\n",
    "    file_dict_o = dict(zip(annotation_file.file, annotation_file.sample_name))\n",
    "    original_file_dict = dict()\n",
    "    for old_key in file_dict_o.keys():\n",
    "        new_key = old_key.split(\"\\\\\")[-1].split(\".\")[0]\n",
    "        original_file_dict[new_key] = file_dict_o[old_key]\n",
    "\n",
    "    c_annotation_path = c_folder + r\"\\experiment_annotation.tsv\"\n",
    "    annotation_file = pd.read_csv(c_annotation_path, sep = '\\t')\n",
    "    file_dict_c = dict(zip(annotation_file.file, annotation_file.sample_name))\n",
    "    censored_file_dict = dict()\n",
    "    for old_key in file_dict_c.keys():\n",
    "        new_key = old_key.replace(\"-censored\", \"\").split(\"\\\\\")[-1].split(\".\")[0]\n",
    "        censored_file_dict[new_key] = file_dict_c[old_key]\n",
    "\n",
    "    # Read in the censored psms\n",
    "    censored_psms = pd.read_csv(censored_psm_path, sep = '\\t')\n",
    "    censored_psms[\"File Name\"] = censored_psms[\"Spectrum File\"].apply(lambda x: x.replace(\"-censored\", \"\").replace(\"interact-\", \"\").split(\"\\\\\")[-1].split(\".\")[0])\n",
    "    # The mods got screwed up during writing the psms, so we fix that here\n",
    "    censored_psms[\"Modified Peptide\"] = censored_psms.apply(lambda x: x[\"Modified Peptide\"].replace(\"43\", \"42.0106\"), axis = 1)\n",
    "    censored_psms[\"Modified Peptide\"] = censored_psms.apply(lambda x: x[\"Modified Peptide\"].replace(\"147\", \"15.9949\"), axis = 1)\n",
    "    censored_psms[\"Modified Peptide\"] = censored_psms.apply(lambda x: x[\"Modified Peptide\"].replace(\"C\", \"C[57.0215]\"), axis = 1)\n",
    "\n",
    "    # Read in the original ions\n",
    "    o_ion_path = o_folder + r\"\\combined_ion.tsv\"\n",
    "    original_ions = pd.read_csv(o_ion_path, sep = '\\t')\n",
    "\n",
    "    info_cols = original_ions.columns[[0, 1, 7,8,11,12,13,14,15,16, 17]].to_list()\n",
    "\n",
    "    match_cols = [col for col in original_ions.columns if \"Match Type\" in col]\n",
    "    intensity_cols = [col for col in original_ions.columns if \"Intensity\" in col]\n",
    "    rt_cols = [col for col in original_ions.columns if \"Apex Retention Time\" in col]\n",
    "\n",
    "    match_cols = [match_cols[i] for i in human_rep_indices]\n",
    "    rt_cols = [rt_cols[i] for i in human_rep_indices]\n",
    "    intensity_cols = [intensity_cols[i] for i in human_rep_indices]\n",
    "\n",
    "    original_ions = original_ions[info_cols + match_cols + rt_cols + intensity_cols]\n",
    "\n",
    "    ion_melt = pd.melt(original_ions, id_vars = info_cols + match_cols + intensity_cols, value_vars=rt_cols, var_name=\"RT Column\", value_name=\"RT\")\n",
    "    ion_melt[\"Match Type\"] = ion_melt.apply(get_match_type, axis = 1)\n",
    "    ion_melt[\"Intensity\"] = ion_melt.apply(get_intensity, axis = 1)\n",
    "    ion_melt.drop(match_cols + intensity_cols, axis = 1, inplace = True)\n",
    "    ion_melt = ion_melt.loc[ion_melt[\"Match Type\"] == \"MS/MS\"]\n",
    "\n",
    "    original_file_rev_dict = dict((v, k) for k, v in original_file_dict.items())\n",
    "    ion_melt[\"File Name\"] = ion_melt.apply(lambda x: original_file_rev_dict[x[\"RT Column\"].split(\" \")[0]], axis = 1)\n",
    "    o_ion_merge = pd.merge(left = ion_melt, right=censored_psms[[\"File Name\", \"Modified Peptide\"]],\n",
    "                            how = \"inner\", left_on = [\"File Name\", \"Modified Sequence\"], right_on = [\"File Name\", \"Modified Peptide\"])\n",
    "    o_ion_merge = o_ion_merge.sort_values(\"Intensity\", ascending=False).groupby([\"Modified Sequence\", \"File Name\"], as_index = False).first() # Keep only the highest intensity charge state\n",
    "\n",
    "\n",
    "    #Read in the ions derived from the censored data\n",
    "    # melt it to have one row per peak (need to group rt, intensity, etc.)\n",
    "    c_ion_path = c_folder + r\"\\combined_ion.tsv\"\n",
    "    censored_ions = pd.read_csv(c_ion_path, sep = '\\t')\n",
    "    info_cols = censored_ions.columns[[0, 1, 7,8,11,12,13,14,15,16, 17]].to_list()\n",
    "\n",
    "    match_cols = [col for col in censored_ions.columns if \"Match Type\" in col]\n",
    "    intensity_cols = [col for col in censored_ions.columns if \"Intensity\" in col]\n",
    "    rt_cols = [col for col in censored_ions.columns if \"Apex Retention Time\" in col]\n",
    "\n",
    "    match_cols = [match_cols[i] for i in human_rep_indices_censored]\n",
    "    rt_cols = [rt_cols[i] for i in human_rep_indices_censored]\n",
    "    intensity_cols = [intensity_cols[i] for i in human_rep_indices_censored]\n",
    "\n",
    "    censored_ions = censored_ions[info_cols + match_cols + rt_cols + intensity_cols]\n",
    "\n",
    "    c_ion_melt = pd.melt(censored_ions, id_vars = info_cols + match_cols + intensity_cols, value_vars=rt_cols, var_name=\"RT Column\", value_name=\"RT\")\n",
    "    c_ion_melt[\"Match Type\"] = c_ion_melt.apply(get_match_type, axis = 1)\n",
    "    c_ion_melt[\"Intensity\"] = c_ion_melt.apply(get_intensity, axis = 1)\n",
    "    c_ion_melt.drop(match_cols + intensity_cols, axis = 1, inplace = True)\n",
    "\n",
    "    censored_file_rev_dict = dict((v, k) for k, v in censored_file_dict.items())\n",
    "    c_ion_melt[\"File Name\"] = c_ion_melt.apply(lambda x: censored_file_rev_dict[x[\"RT Column\"].split(\" \")[0]], axis = 1)\n",
    "    c_ion_melt = c_ion_melt.loc[c_ion_melt[\"Match Type\"] == \"MBR\"]\n",
    "    c_ion_merge = pd.merge(left = c_ion_melt, right=censored_psms[[\"File Name\", \"Modified Peptide\"]],\n",
    "                            how = \"right\", left_on = [\"File Name\", \"Modified Sequence\"], right_on = [\"File Name\", \"Modified Peptide\"])\n",
    "    c_ion_merge = c_ion_merge.sort_values(\"Intensity\", ascending=False).groupby([\"Modified Sequence\", \"File Name\"], as_index = False).first() # Keep only the highest intensity charge state\n",
    "\n",
    "    # Calculate the sensitivity (number of MBR ions / number of ions detected w/ MSMS)\n",
    "    ion_comp = pd.merge(o_ion_merge, c_ion_merge, on = [\"Modified Peptide\", \"File Name\"], suffixes = (\"_o\", \"_c\"), how = \"inner\")\n",
    "    ion_comp = ion_comp[ion_comp[\"RT_o\"].notna() & ion_comp[\"RT_c\"].notna()]\n",
    "    ion_comp[\"RT_diff\"] = abs(ion_comp[\"RT_o\"] - ion_comp[\"RT_c\"])\n",
    "    ion_comp.sort_values(\"RT_diff\", ascending=True, inplace=True)\n",
    "    ion_comp = ion_comp.drop_duplicates(subset = [\"Modified Peptide\", \"File Name\"], keep = 'first')\n",
    "    specificity = sum(ion_comp[\"RT_diff\"] > rt_diff_threshold) / ion_comp.shape[0]\n",
    "\n",
    "    ion_diffs = ion_comp.loc[ion_comp[\"RT_diff\"] > rt_diff_threshold]\n",
    "    pct_diffs = ion_diffs.apply(lambda x: 100 * abs(x[\"Intensity_o\"] - x[\"Intensity_c\"]) /\n",
    "                                             ((x[\"Intensity_o\"] + x[\"Intensity_c\"]) / 2.0), axis = 1)\n",
    "    \n",
    "\n",
    "    o_ion_merge.drop_duplicates(subset = [\"Modified Peptide\", \"File Name\"], inplace = True)\n",
    "    c_ion_merge.drop_duplicates(subset = [\"Modified Peptide\", \"File Name\"], inplace = True)\n",
    "    number_of_ions_original = o_ion_merge.shape[0] - o_ion_merge[\"RT\"].isna().sum()\n",
    "    number_of_ions_censored = c_ion_merge.shape[0] - c_ion_merge[\"RT\"].isna().sum()\n",
    "    sensitivity = number_of_ions_censored / number_of_ions_original\n",
    "\n",
    "    results = dict(sensitivity = sensitivity, specificity = specificity, intensity_diff = pct_diffs.mean())\n",
    "    return results\n",
    "\n",
    "\n",
    "# combined_ion parsing functions\n",
    "def get_match_type(row):\n",
    "    return(row[row[\"RT Column\"].split(\" \")[0] + \" Match Type\"])\n",
    "\n",
    "def get_intensity(row):\n",
    "    return(row[row[\"RT Column\"].split(\" \")[0] + \" Intensity\"])\n",
    "\n",
    "def count_ions_by_species(path_to_combined_ion, match_indices = [0, 1, 2, 3, 4, 5, 6 ], foreign_species = \"YEAST\"):\n",
    "    entrapment_species = \"_ENT\"\n",
    "\n",
    "    # Read in combined ions\n",
    "    ions = pd.read_csv(path_to_combined_ion, sep = '\\t')\n",
    "    # Keep informative columns (species, peptide, etc.) and \"... Match Type\" columns\n",
    "    match_cols = [col for col in ions.columns if \"Match Type\" in col]\n",
    "    # WARNING - This line is experiment dependent and should be changed based on the samples you wish to analyze\n",
    "    match_cols = [match_cols[i] for i in match_indices] #Remove the two mixed proteome samples \n",
    "    info_cols = ions.columns[[0, 1, 7,8,11,12,13,14,15,16, 17]].to_list()\n",
    "    ions = ions[info_cols + match_cols]\n",
    "    # us melt to create one row per sample match type\n",
    "    ion_mbr = ions.melt(info_cols)\n",
    "\n",
    "    foreign_msms = ion_mbr.loc[(ion_mbr[\"Entry Name\"].str.contains(foreign_species)) & (ion_mbr[\"value\"] == \"MS/MS\")]\n",
    "    foreign_msms_seqs = set(foreign_msms[\"Modified Sequence\"].tolist())\n",
    "    \n",
    "    ion_msms = ion_mbr.loc[ion_mbr.value == \"MS/MS\"]\n",
    "    ion_msms = ion_msms.loc[(ion_msms[\"Entry Name\"].str.contains(\"HUMAN\")) | (ion_msms[\"Entry Name\"].str.contains(foreign_species)) | (ion_msms[\"Entry Name\"].str.contains(entrapment_species))]\n",
    "    msms_count = ion_msms.shape[0]\n",
    "\n",
    "    # keep only the MBR match types\n",
    "    ion_mbr = ion_mbr.loc[ion_mbr.value == \"MBR\"]\n",
    "\n",
    "    ion_mbr.drop_duplicates(subset = [\"Modified Sequence\", \"variable\"], inplace = True)\n",
    "\n",
    "    # Count the number of ions from each species\n",
    "    human_ions = ion_mbr.loc[ion_mbr[\"Entry Name\"].str.contains(\"HUMAN\")]\n",
    "    real_human_ions = human_ions.loc[~human_ions[\"Peptide Sequence\"].isin(entrapment_seqs_set)]\n",
    "    entrapment_ions = human_ions.loc[human_ions[\"Peptide Sequence\"].isin(entrapment_seqs_set)]\n",
    "    human_ion_count = real_human_ions.shape[0]\n",
    "    arath_ion_count = entrapment_ions.shape[0]\n",
    "\n",
    "    foreign_ions = ion_mbr.loc[ion_mbr[\"Entry Name\"].str.contains(foreign_species)]\n",
    "    foreign_ions[\"Mapped Proteins\"] = foreign_ions[\"Mapped Proteins\"].fillna('')\n",
    "    foreign_ions = foreign_ions.loc[~foreign_ions[\"Mapped Proteins\"].str.contains(\"HUMAN\")]\n",
    "    foreign_ion_count = foreign_ions.shape[0]\n",
    "\n",
    "    foreign_double_count = sum(foreign_ions[\"Modified Sequence\"].isin(foreign_msms_seqs))\n",
    "    foreign_ion_count = foreign_ion_count - foreign_double_count\n",
    "\n",
    "\n",
    "\n",
    "    count_dict = dict(\n",
    "        {\n",
    "            \"human\" : human_ion_count,\n",
    "            \"foreign\" : foreign_ion_count,\n",
    "            \"arabida\" : arath_ion_count,\n",
    "            \"total\" : human_ion_count + foreign_ion_count + arath_ion_count,\n",
    "            \"msms_count\" : msms_count,\n",
    "            \"mbr_count\" : ion_mbr.shape[0]\n",
    "        })\n",
    "    return(count_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_fdp_maxquant(path, foreign_species = \"Yeast\", human_file_pattern = \"_human_90min_\"):\n",
    "    \n",
    "    ion_count = count_ions_by_species_mq(path, foreign_species, human_file_pattern)\n",
    "\n",
    "    scaled_arabida = ion_count[\"arabida\"] * scaling_factor\n",
    "        \n",
    "    ion_count[\"scaled_arabida\"] = scaled_arabida\n",
    "\n",
    "    ion_count[\"fper\"] = 100 * ion_count[\"foreign\"]  / ion_count[\"total\"]\n",
    "    ion_count[\"nper\"] = 0\n",
    "    ion_count[\"fder\"] = (100-ion_count[\"fper\"]) * ion_count[\"scaled_arabida\"]  / ion_count[\"total\"]\n",
    "    ion_count[\"false_discovery_proportion\"] = ion_count[\"fper\"] + ion_count[\"nper\"] + ion_count[\"fder\"]\n",
    "\n",
    "    return ion_count\n",
    "\n",
    "def count_ions_by_species_mq(evidence_path = r\"D:\\Gygi_TwoProteomeData\\combined_MaxQuant_Gygi\\txt\\evidence.txt\", foreign_species = \"Yeast\",\n",
    "                             human_file_pattern = \"_human_90min_\"):\n",
    "    \n",
    "    evidence = pd.read_csv(evidence_path, sep = '\\t')\n",
    "    evidence = evidence.loc[(evidence[\"Reverse\"] != \"+\") & (evidence[\"Raw file\"].str.contains(human_file_pattern))]\n",
    "    evidence[\"Proteins\"] = evidence[\"Proteins\"].astype(str)\n",
    "    evidence[\"Organism\"] = evidence.apply(lambda x: ';'.join([get_species(protein) for protein in x['Proteins'].split(';')]), axis = 1)\n",
    "    \n",
    "    msms_foreign = evidence.loc[(evidence[\"MS/MS count\"] > 0) & (evidence[\"Organism\"].str.contains(foreign_species))]\n",
    "    msms_foreign_seq = set(msms_foreign[\"Modified sequence\"].tolist())\n",
    "\n",
    "    pip = evidence.loc[evidence[\"Match score\"].isna() == False]\n",
    "    pip = pip.loc[pip[\"Raw file\"].str.contains(human_file_pattern)]\n",
    "\n",
    "    human_ions = pip.loc[pip[\"Organism\"].str.contains(\"Human\")]\n",
    "    human_ion_real = human_ions.loc[~human_ions[\"Sequence\"].isin(entrapment_seqs_set)]  \n",
    "    entrapment_ions = human_ions.loc[human_ions[\"Sequence\"].isin(entrapment_seqs_set)]  \n",
    "    human_ion_count = human_ion_real.shape[0]\n",
    "    arath_ion_count = entrapment_ions.shape[0]\n",
    "\n",
    "    foreign_ions = pip.loc[pip[\"Organism\"].str.contains(foreign_species)]\n",
    "    foreign_ions = foreign_ions.loc[~foreign_ions[\"Organism\"].str.contains(\"Human\")]\n",
    "    foreign_ion_count = foreign_ions.shape[0]\n",
    "\n",
    "    foreign_ion_double_count = sum(foreign_ions[\"Modified sequence\"].isin(msms_foreign_seq))\n",
    "    foreign_ion_count = foreign_ion_count - foreign_ion_double_count\n",
    "\n",
    "    #arath_ions = pip.loc[(pip[\"Organism\"].str.contains(\"Arabida\")) | (pip[\"Proteins\"].str.contains(\"ENTRAPMENT\"))]  \n",
    "    #arath_ions = arath_ions.loc[~arath_ions[\"Organism\"].str.contains(\"Human\")]\n",
    "    #arath_ions = arath_ions.loc[~arath_ions[\"Organism\"].str.contains(foreign_species)]\n",
    "    #arath_ion_count = arath_ions.shape[0]\n",
    "\n",
    "    count_dict = dict(\n",
    "        {\n",
    "            \"human\" : human_ion_count,\n",
    "            \"foreign\" : foreign_ion_count,\n",
    "            \"arabida\" : arath_ion_count,\n",
    "            \"total\" : sum([human_ion_count, foreign_ion_count, arath_ion_count])\n",
    "        })\n",
    "    return count_dict\n",
    "\n",
    "def get_species(accession):\n",
    "    species = accession_dict.get(accession, 'Other')\n",
    "    return species"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gygi RT Threshold:  0.9\n",
      "Kelly RT Threshold:  0.4\n",
      "Inhouse RT Threshold:  0.6\n"
     ]
    }
   ],
   "source": [
    "# RT thresholds for native peak error calculations\n",
    "\n",
    "gygi_rt = 90 * 0.01\n",
    "inhouse_rt = 60 * 0.01\n",
    "kelly_rt = 40 * 0.01\n",
    "\n",
    "print(\"Gygi RT Threshold: \", gygi_rt)\n",
    "print(\"Kelly RT Threshold: \", kelly_rt)\n",
    "print(\"Inhouse RT Threshold: \", inhouse_rt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1976879677.py:5: DtypeWarning: Columns (9,10,11,12,13) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1976879677.py:79: DtypeWarning: Columns (9,10,11,12,13) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  original_peaks = pd.read_csv(old_peak_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1976879677.py:87: DtypeWarning: Columns (9,10,11,12,13) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  new_peaks = pd.read_csv(new_peak_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1976879677.py:5: DtypeWarning: Columns (9,10,11,12,13) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1976879677.py:79: DtypeWarning: Columns (9,10,11,12,13) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  original_peaks = pd.read_csv(old_peak_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1976879677.py:87: DtypeWarning: Columns (9,10,11,12,13) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  new_peaks = pd.read_csv(new_peak_path, sep = '\\t')\n"
     ]
    }
   ],
   "source": [
    "gygi_flashv1 = get_fdp_old_flash(o_peaks_path = r\"D:\\GygiTwoProteome_PXD014415\\FlashLFQ_CurRel_PepQ_ConcatenatedDB\\QuantifiedPeaks.tsv\",\n",
    "                          c_peaks_path = r\"D:\\GygiTwoProteome_PXD014415\\CensoredData_FlashLFQ_CurRel_PepQ_ConcatenatedDB\\QuantifiedPeaks.tsv\",\n",
    "                          censored_psm_path = r\"D:\\GygiTwoProteome_PXD014415\\MsConvertmzMLs\\MM_NewPep_CensoredMzmls\\CensoredPsms.psmtsv\",\n",
    "                          human_file_pattern=\"human_90\", foreign_species='Yeast', rt_delta=gygi_rt)\n",
    "\n",
    "inhouse_flashv1 = get_fdp_old_flash(o_peaks_path = r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\Human_FlashLFQ_CurRel_PepQ_ConcatenatedDB\\QuantifiedPeaks.tsv\",\n",
    "                          c_peaks_path = r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\CensoredHuman_FlashLFQ_CurRel_PepQ_ConcatenatedDB\\QuantifiedPeaks.tsv\",\n",
    "                          censored_psm_path = r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\RawData\\MM_CensoredMzml_8_3_24\\CensoredPsms.psmtsv\",\n",
    "                          human_file_pattern=\"Human_C18\", foreign_species='Ecoli', rt_delta=inhouse_rt)\n",
    "\n",
    "kelly_flashv1= get_fdp_old_flash(o_peaks_path = r\"D:\\Kelly_TwoProteomeData\\FlashLFQ_CurRel_PepQ_ConcatenatedDB\\QuantifiedPeaks.tsv\",\n",
    "                          c_peaks_path = r\"D:\\Kelly_TwoProteomeData\\CensoredData_FlashLFQ_CurRel_PepQ_ConcatenatedDB\\QuantifiedPeaks.tsv\",\n",
    "                          censored_psm_path = r\"D:\\Kelly_TwoProteomeData\\MsConvertMzMls\\MM_Censored_8_2_24\\CensoredPsms.psmtsv\",\n",
    "                          human_file_pattern=\"_1x02nguL_\", foreign_species='Yeast', rt_delta=kelly_rt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:4: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:101: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  original_peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:111: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  new_peaks = pd.read_csv(c_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:4: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:101: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  original_peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:111: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  new_peaks = pd.read_csv(c_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:4: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:101: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  original_peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:111: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  new_peaks = pd.read_csv(c_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:4: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:101: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  original_peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:111: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  new_peaks = pd.read_csv(c_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:4: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:101: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  original_peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:111: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  new_peaks = pd.read_csv(c_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:4: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:101: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  original_peaks = pd.read_csv(o_peaks_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\1540678269.py:111: DtypeWarning: Columns (10,11,12,13,14) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  new_peaks = pd.read_csv(c_peaks_path, sep = '\\t')\n"
     ]
    }
   ],
   "source": [
    "# Current FlashLFQ\n",
    "gygi_censored_psms = r\"D:\\GygiTwoProteome_PXD014415\\MsConvertmzMLs\\MM_NewPep_CensoredMzmls\\CensoredPsms.psmtsv\"\n",
    "\n",
    "gygi_dd_1 = get_fdp_flash(o_peaks_path = r\"D:\\GygiTwoProteome_PXD014415\\FlashLFQ_7772_DonorPepQ_02\\QuantifiedPeaks.tsv\",\n",
    "                          c_peaks_path = r\"D:\\GygiTwoProteome_PXD014415\\CensoredData_FlashLFQ_7772_DonorPepQ_02\\QuantifiedPeaks.tsv\",\n",
    "                          censored_psm_path = gygi_censored_psms,\n",
    "                          human_file_pattern=\"human_90\", foreign_species='Saccharomyces cerevisiae', q_value=0.01, rt_delta=gygi_rt)\n",
    "\n",
    "gygi_dd_2p5 = get_fdp_flash(o_peaks_path = r\"D:\\GygiTwoProteome_PXD014415\\FlashLFQ_7772_DonorPepQ_05\\QuantifiedPeaks.tsv\",\n",
    "                            c_peaks_path = r\"D:\\GygiTwoProteome_PXD014415\\CensoredData_FlashLFQ_7772_DonorPepQ_05\\QuantifiedPeaks.tsv\",\n",
    "                            censored_psm_path = gygi_censored_psms,\n",
    "                            human_file_pattern=\"human_90\", foreign_species='Saccharomyces cerevisiae', q_value=0.025, rt_delta=gygi_rt)\n",
    "\n",
    "gygi_dd_5 = get_fdp_flash(o_peaks_path = r\"D:\\GygiTwoProteome_PXD014415\\FlashLFQ_7772_DonorPepQ_1\\QuantifiedPeaks.tsv\",\n",
    "                          c_peaks_path = r\"D:\\GygiTwoProteome_PXD014415\\CensoredData_FlashLFQ_7772_DonorPepQ_1\\QuantifiedPeaks.tsv\",\n",
    "                          censored_psm_path = gygi_censored_psms,\n",
    "                          human_file_pattern=\"human_90\", foreign_species='Saccharomyces cerevisiae', q_value=0.05, rt_delta=gygi_rt)\n",
    "\n",
    "inhouse_censored_psms = r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\RawData\\MM_CensoredMzml_8_3_24\\CensoredPsms.psmtsv\"\n",
    "\n",
    "inhouse_dd_1 = get_fdp_flash(o_peaks_path = r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\Human_FlashLFQ_7772_DonorPepQ_02\\QuantifiedPeaks.tsv\",\n",
    "                          c_peaks_path = r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\CensoredHuman_FlashLFQ_7772_DonorPepQ_02\\QuantifiedPeaks.tsv\",\n",
    "                          censored_psm_path = inhouse_censored_psms,\n",
    "                          human_file_pattern=\"Human_C18\", foreign_species='Escherichia coli', q_value=0.01, rt_delta=inhouse_rt)\n",
    "\n",
    "inhouse_dd_2p5 = get_fdp_flash(o_peaks_path = r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\Human_FlashLFQ_7772_DonorPepQ_05\\QuantifiedPeaks.tsv\",\n",
    "                          c_peaks_path = r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\CensoredHuman_FlashLFQ_7772_DonorPepQ_05\\QuantifiedPeaks.tsv\",\n",
    "                          censored_psm_path = inhouse_censored_psms,\n",
    "                          human_file_pattern=\"Human_C18\", foreign_species='Escherichia coli', q_value=0.025, rt_delta=inhouse_rt)\n",
    "\n",
    "inhouse_dd_5 = get_fdp_flash(o_peaks_path = r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\Human_FlashLFQ_7772_DonorPepQ_1\\QuantifiedPeaks.tsv\",\n",
    "                          c_peaks_path = r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\CensoredHuman_FlashLFQ_7772_DonorPepQ_1\\QuantifiedPeaks.tsv\",\n",
    "                          censored_psm_path = inhouse_censored_psms,\n",
    "                          human_file_pattern=\"Human_C18\", foreign_species='Escherichia coli', q_value=0.05, rt_delta=inhouse_rt)\n",
    "\n",
    "kelly_censored_psms = r\"D:\\Kelly_TwoProteomeData\\MsConvertMzMls\\MM_Censored_8_2_24\\CensoredPsms.psmtsv\"\n",
    "\n",
    "kelly_dd_1 = get_fdp_flash(o_peaks_path = r\"D:\\Kelly_TwoProteomeData\\FlashLFQ_7772_DonorPepQ_02\\QuantifiedPeaks.tsv\",\n",
    "                          c_peaks_path = r\"D:\\Kelly_TwoProteomeData\\CensoredData_FlashLFQ_7772_DonorPepQ_02\\QuantifiedPeaks.tsv\",\n",
    "                          censored_psm_path = kelly_censored_psms,\n",
    "                          human_file_pattern=\"_1x02nguL_\", foreign_species='Saccharomyces cerevisiae', q_value=0.01, rt_delta=kelly_rt)\n",
    "\n",
    "kelly_dd_2p5 = get_fdp_flash(o_peaks_path = r\"D:\\Kelly_TwoProteomeData\\FlashLFQ_7772_DonorPepQ_05\\QuantifiedPeaks.tsv\",\n",
    "                          c_peaks_path = r\"D:\\Kelly_TwoProteomeData\\CensoredData_FlashLFQ_7772_DonorPepQ_05\\QuantifiedPeaks.tsv\",\n",
    "                          censored_psm_path = kelly_censored_psms,\n",
    "                          human_file_pattern=\"_1x02nguL_\", foreign_species='Saccharomyces cerevisiae', q_value=0.025, rt_delta=kelly_rt)\n",
    "\n",
    "\n",
    "kelly_dd_5 = get_fdp_flash(o_peaks_path = r\"D:\\Kelly_TwoProteomeData\\FlashLFQ_7772_DonorPepQ_1\\QuantifiedPeaks.tsv\",\n",
    "                          c_peaks_path = r\"D:\\Kelly_TwoProteomeData\\CensoredData_FlashLFQ_7772_DonorPepQ_1\\QuantifiedPeaks.tsv\",\n",
    "                          censored_psm_path = kelly_censored_psms,\n",
    "                          human_file_pattern=\"_1x02nguL_\", foreign_species='Saccharomyces cerevisiae', q_value=0.05, rt_delta=kelly_rt)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gygi_iq_1 = get_fdp_fragger(o_folder = r\"D:\\GygiTwoProteome_PXD014415\\MsConvertmzMLs\\IonQuant_1Percent_50PercentEntrapmentDb\",\n",
    "                                c_folder= r\"D:\\GygiTwoProteome_PXD014415\\MsConvertmzMLs\\CensoredData_IonQuant_1Percent_50PercentEntrapmentDb\",\n",
    "                                censored_psm_path=r\"D:\\GygiTwoProteome_PXD014415\\MsConvertmzMLs\\CensoredFiles_Fragger_8_15_24\\CensoredPsms.tsv\",\n",
    "                                human_file_pattern=\"human_90\",\n",
    "                                rt_diff_threshold=gygi_rt)\n",
    "\n",
    "gygi_iq_2p5 = get_fdp_fragger(o_folder = r\"D:\\GygiTwoProteome_PXD014415\\MsConvertmzMLs\\IonQuant_2p5Percent_50PercentEntrapmentDb\",\n",
    "                                c_folder= r\"D:\\GygiTwoProteome_PXD014415\\MsConvertmzMLs\\CensoredData_IonQuant_2p5Percent_50PercentEntrapmentDb\",\n",
    "                                censored_psm_path=r\"D:\\GygiTwoProteome_PXD014415\\MsConvertmzMLs\\CensoredFiles_Fragger_8_15_24\\CensoredPsms.tsv\",\n",
    "                                human_file_pattern=\"human_90\",\n",
    "                                rt_diff_threshold=gygi_rt) \n",
    "\n",
    "gygi_iq_5 = get_fdp_fragger(o_folder = r\"D:\\GygiTwoProteome_PXD014415\\MsConvertmzMLs\\IonQuant_5Percent_50PercentEntrapmentDb\",\n",
    "                                c_folder= r\"D:\\GygiTwoProteome_PXD014415\\MsConvertmzMLs\\CensoredData_IonQuant_5Percent_50PercentEntrapmentDb\",\n",
    "                                censored_psm_path=r\"D:\\GygiTwoProteome_PXD014415\\MsConvertmzMLs\\CensoredFiles_Fragger_8_15_24\\CensoredPsms.tsv\",\n",
    "                                human_file_pattern=\"human_90\",\n",
    "                                rt_diff_threshold=gygi_rt)\n",
    "\n",
    "inhouse_iq_1 = get_fdp_fragger(o_folder = r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\RawData\\MzMls\\IonQuant_1Percent_50PercentEntrapmentDb\", \n",
    "                            c_folder= r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\RawData\\MzMls\\CensoredData_IonQuant_1Percent_50PercentEntrapmentDb\",\n",
    "                            censored_psm_path=r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\RawData\\CensoredFiles_Fragger_8_16_24\\CensoredPsms.tsv\", \n",
    "                            human_file_pattern=\"_Human_C18_\",\n",
    "                            foreign_species=\"ECOLI\",\n",
    "                            rt_diff_threshold=inhouse_rt)\n",
    "\n",
    "inhouse_iq_2p5 = get_fdp_fragger(o_folder = r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\RawData\\MzMls\\IonQuant_2p5Percent_50PercentEntrapmentDb\", \n",
    "                            c_folder= r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\RawData\\MzMls\\CensoredData_IonQuant_2p5Percent_50PercentEntrapmentDb\",\n",
    "                            censored_psm_path=r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\RawData\\CensoredFiles_Fragger_8_16_24\\CensoredPsms.tsv\", \n",
    "                            human_file_pattern=\"_Human_C18_\",\n",
    "                            foreign_species=\"ECOLI\",\n",
    "                            rt_diff_threshold=inhouse_rt)\n",
    "\n",
    "inhouse_iq_5 = get_fdp_fragger(o_folder = r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\RawData\\MzMls\\IonQuant_5Percent_50PercentEntrapmentDb\", \n",
    "                            c_folder= r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\RawData\\MzMls\\CensoredData_IonQuant_5Percent_50PercentEntrapmentDb\",\n",
    "                            censored_psm_path=r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\RawData\\CensoredFiles_Fragger_8_16_24\\CensoredPsms.tsv\", \n",
    "                            human_file_pattern=\"_Human_C18_\",\n",
    "                            foreign_species=\"ECOLI\",\n",
    "                            rt_diff_threshold=inhouse_rt)\n",
    "\n",
    "kelly_iq_1 = get_fdp_fragger(o_folder = r\"D:\\Kelly_TwoProteomeData\\MsConvertMzMls\\LibrarySettings_1Percent\", \n",
    "                            c_folder= r\"D:\\Kelly_TwoProteomeData\\MsConvertMzMls\\LibrarySettings_CensoredData_1Percent\",\n",
    "                            censored_psm_path=r\"D:\\Kelly_TwoProteomeData\\MsConvertMzMls\\CensoredFiles_Fragger_8_12_24\\CensoredPsms.tsv\", \n",
    "                            human_file_pattern=\"_1x02nguL_\",\n",
    "                            rt_diff_threshold=kelly_rt)\n",
    "\n",
    "kelly_iq_2p5 = get_fdp_fragger(o_folder = r\"D:\\Kelly_TwoProteomeData\\MsConvertMzMls\\LibrarySettings_2p5Percent_2\", \n",
    "                            c_folder= r\"D:\\Kelly_TwoProteomeData\\MsConvertMzMls\\LibrarySettings_CensoredData_2p5Percent\",\n",
    "                            censored_psm_path=r\"D:\\Kelly_TwoProteomeData\\MsConvertMzMls\\CensoredFiles_Fragger_8_12_24\\CensoredPsms.tsv\", \n",
    "                            human_file_pattern=\"_1x02nguL_\",\n",
    "                            rt_diff_threshold=kelly_rt)\n",
    "\n",
    "kelly_iq_5 = get_fdp_fragger(o_folder = r\"D:\\Kelly_TwoProteomeData\\MsConvertMzMls\\LibrarySettings_5Percent_2\", \n",
    "                            c_folder= r\"D:\\Kelly_TwoProteomeData\\MsConvertMzMls\\LibrarySettings_CensoredData_5Percent\",\n",
    "                            censored_psm_path=r\"D:\\Kelly_TwoProteomeData\\MsConvertMzMls\\CensoredFiles_Fragger_8_12_24\\CensoredPsms.tsv\", \n",
    "                            human_file_pattern=\"_1x02nguL_\",\n",
    "                            rt_diff_threshold=kelly_rt)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\4547196.py:19: DtypeWarning: Columns (51,52,59) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  evidence = pd.read_csv(evidence_path, sep = '\\t')\n",
      "C:\\Users\\Alex\\AppData\\Local\\Temp\\ipykernel_40240\\4547196.py:19: DtypeWarning: Columns (49) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  evidence = pd.read_csv(evidence_path, sep = '\\t')\n"
     ]
    }
   ],
   "source": [
    "gygi_mq = get_fdp_maxquant(r\"D:\\GygiTwoProteome_PXD014415\\combined_Gygi_8_16_Concatenated\\txt\\evidence.txt\", \"Yeast\", human_file_pattern=\"human_90\")\n",
    "\n",
    "inhouse_mq = get_fdp_maxquant(r\"D:\\Human_Ecoli_TwoProteome_60minGradient\\combined_Inhouse_8_16_Concatenated\\txt\\evidence.txt\", \"Ecoli\", human_file_pattern=\"_Human_C18_\")\n",
    "\n",
    "kelly_mq = get_fdp_maxquant(r\"D:\\Kelly_TwoProteomeData\\combined_Kelly_8_15_Concatenated\\txt\\evidence.txt\", \"Yeast\", human_file_pattern=\"_1x02nguL_\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fdp_df = pd.DataFrame(\n",
    "    {\n",
    "        \"Gygi_Flash_v1\" : pd.Series(gygi_flashv1),\n",
    "        \"Gygi_Flash_v2_1\" : pd.Series(gygi_dd_1),\n",
    "        \"Gygi_Flash_v2_2.5\" : pd.Series(gygi_dd_2p5),\n",
    "        \"Gygi_Flash_v2_5\" : pd.Series(gygi_dd_5),\n",
    "        \"Gygi_IonQuant_1\" : pd.Series(gygi_iq_1),\n",
    "        \"Gygi_IonQuant_2.5\" : pd.Series(gygi_iq_2p5),\n",
    "        \"Gygi_IonQuant_5\" : pd.Series(gygi_iq_5),\n",
    "        \"Gygi_MaxQuant\" : pd.Series(gygi_mq),\n",
    "\n",
    "        \"Kelly_Flash_v1\" : pd.Series(kelly_flashv1),\n",
    "        \"Kelly_Flash_v2_1\" : pd.Series(kelly_dd_1),\n",
    "        \"Kelly_Flash_v2_2.5\" : pd.Series(kelly_dd_2p5),\n",
    "        \"Kelly_Flash_v2_5\" : pd.Series(kelly_dd_5),\n",
    "        \"Kelly_IonQuant_1\" : pd.Series(kelly_iq_1),\n",
    "        \"Kelly_IonQuant_2.5\" : pd.Series(kelly_iq_2p5),\n",
    "        \"Kelly_IonQuant_5\" : pd.Series(kelly_iq_5),\n",
    "        \"Kelly_MaxQuant\" : pd.Series(kelly_mq),\n",
    "\n",
    "        \"Inhouse_Flash_v1\" : pd.Series(inhouse_flashv1),\n",
    "        \"Inhouse_Flash_v2_1\" : pd.Series(inhouse_dd_1),\n",
    "        \"Inhouse_Flash_v2_2.5\" : pd.Series(inhouse_dd_2p5),\n",
    "        \"Inhouse_Flash_v2_5\" : pd.Series(inhouse_dd_5),\n",
    "        \"Inhouse_IonQuant_1\" : pd.Series(inhouse_iq_1),\n",
    "        \"Inhouse_IonQuant_2.5\" : pd.Series(inhouse_iq_2p5),\n",
    "        \"Inhouse_IonQuant_5\" : pd.Series(inhouse_iq_5),\n",
    "        \"Inhouse_MaxQuant\" : pd.Series(inhouse_mq)\n",
    "    }\n",
    ")\n",
    "\n",
    "fdp_df = fdp_df.transpose()\n",
    "fdp_df[\"software\"] = fdp_df.index\n",
    "\n",
    "fdp_df.to_csv(r\"C:\\Users\\Alex\\Source\\Repos\\MBR_metamorpheus\\R_Files\\FDP_Analysis_Results.tsv\", sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FlashLFQ v1</th>\n",
       "      <th>FlashLFQ v2 1%</th>\n",
       "      <th>FlashLFQ v2 5%</th>\n",
       "      <th>IonQuant 1%</th>\n",
       "      <th>IonQuant 5%</th>\n",
       "      <th>MaxQuant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Native Peak Errors</th>\n",
       "      <td>1040</td>\n",
       "      <td>230</td>\n",
       "      <td>1003</td>\n",
       "      <td>18118</td>\n",
       "      <td>19649</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Foreign Peak Errors</th>\n",
       "      <td>4870</td>\n",
       "      <td>407</td>\n",
       "      <td>4535</td>\n",
       "      <td>1498</td>\n",
       "      <td>6154</td>\n",
       "      <td>21831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>False Detection Errors</th>\n",
       "      <td>15992</td>\n",
       "      <td>572</td>\n",
       "      <td>1680</td>\n",
       "      <td>234</td>\n",
       "      <td>428</td>\n",
       "      <td>1270</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        FlashLFQ v1  FlashLFQ v2 1%  FlashLFQ v2 5%  \\\n",
       "Native Peak Errors             1040             230            1003   \n",
       "Foreign Peak Errors            4870             407            4535   \n",
       "False Detection Errors        15992             572            1680   \n",
       "\n",
       "                        IonQuant 1%  IonQuant 5%  MaxQuant  \n",
       "Native Peak Errors            18118        19649         0  \n",
       "Foreign Peak Errors            1498         6154     21831  \n",
       "False Detection Errors          234          428      1270  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# the following code was adapted from the following SO answer: https://stackoverflow.com/a/47040206\n",
    "gygi_raw = pd.DataFrame(reversed([\n",
    "    [int(7663*2.087), 572, 1680, 234, 428, int(gygi_mq_res[\"arabida\"] * 2.087)],\n",
    "    [4870, 407, 4535, 1498, 6154, int(gygi_mq_res[\"foreign\"])],\n",
    "    [1040, 230, 1003, 18118, 19649, 0]\n",
    "]),\n",
    "columns = ['FlashLFQ v1', 'FlashLFQ v2 1%', 'FlashLFQ v2 5%',  'IonQuant 1%', 'IonQuant 5%', \"MaxQuant\"],\n",
    "index = reversed([\"False Detection Errors\", \"Foreign Peak Errors\", \"Native Peak Errors\"]))\n",
    "gygi_raw"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FlashLFQ v1</th>\n",
       "      <th>FlashLFQ v2 1%</th>\n",
       "      <th>FlashLFQ v2 5%</th>\n",
       "      <th>IonQuant 1%</th>\n",
       "      <th>IonQuant 5%</th>\n",
       "      <th>MaxQuant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Native Peak Errors</th>\n",
       "      <td>228</td>\n",
       "      <td>0</td>\n",
       "      <td>325</td>\n",
       "      <td>26</td>\n",
       "      <td>386</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Foreign Peak Errors</th>\n",
       "      <td>3558</td>\n",
       "      <td>7</td>\n",
       "      <td>1193</td>\n",
       "      <td>66</td>\n",
       "      <td>1124</td>\n",
       "      <td>1127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>False Detection Errors</th>\n",
       "      <td>1613</td>\n",
       "      <td>6</td>\n",
       "      <td>198</td>\n",
       "      <td>2</td>\n",
       "      <td>48</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        FlashLFQ v1  FlashLFQ v2 1%  FlashLFQ v2 5%  \\\n",
       "Native Peak Errors              228               0             325   \n",
       "Foreign Peak Errors            3558               7            1193   \n",
       "False Detection Errors         1613               6             198   \n",
       "\n",
       "                        IonQuant 1%  IonQuant 5%  MaxQuant  \n",
       "Native Peak Errors               26          386         0  \n",
       "Foreign Peak Errors              66         1124      1127  \n",
       "False Detection Errors            2           48       116  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# the following code was adapted from the following SO answer: https://stackoverflow.com/a/47040206\n",
    "kelly_raw = pd.DataFrame(reversed([\n",
    "    [int(773*2.087), 6, 198, int(kelly_1[\"arabida\"] * 2.087), int(kelly_5[\"arabida\"] * 2.087), int(kelly_mq[\"arabida\"] * 2.087)],  #False Detection Errors\n",
    "    [3558, 7, 1193, kelly_1['foreign'], kelly_5['foreign'], int(kelly_mq[\"foreign\"])], # Foreign Peak Errors\n",
    "    [228, 0, 325, 26, 386, 0] # Native Peak Errors\n",
    "]),\n",
    "columns = ['FlashLFQ v1', 'FlashLFQ v2 1%', 'FlashLFQ v2 5%',  'IonQuant 1%', 'IonQuant 5%', \"MaxQuant\"],\n",
    "index = reversed([\"False Detection Errors\", \"Foreign Peak Errors\", \"Native Peak Errors\"]))\n",
    "kelly_raw"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FlashLFQ v1</th>\n",
       "      <th>FlashLFQ v2 1%</th>\n",
       "      <th>FlashLFQ v2 5%</th>\n",
       "      <th>IonQuant 1%</th>\n",
       "      <th>IonQuant 5%</th>\n",
       "      <th>MaxQuant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Native Peak Errors</th>\n",
       "      <td>1239</td>\n",
       "      <td>207</td>\n",
       "      <td>1244</td>\n",
       "      <td>5793</td>\n",
       "      <td>6610</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Foreign Peak Errors</th>\n",
       "      <td>1940</td>\n",
       "      <td>174</td>\n",
       "      <td>1357</td>\n",
       "      <td>383</td>\n",
       "      <td>1080</td>\n",
       "      <td>3987</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>False Detection Errors</th>\n",
       "      <td>12929</td>\n",
       "      <td>438</td>\n",
       "      <td>1200</td>\n",
       "      <td>227</td>\n",
       "      <td>306</td>\n",
       "      <td>1333</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        FlashLFQ v1  FlashLFQ v2 1%  FlashLFQ v2 5%  \\\n",
       "Native Peak Errors             1239             207            1244   \n",
       "Foreign Peak Errors            1940             174            1357   \n",
       "False Detection Errors        12929             438            1200   \n",
       "\n",
       "                        IonQuant 1%  IonQuant 5%  MaxQuant  \n",
       "Native Peak Errors             5793         6610         0  \n",
       "Foreign Peak Errors             383         1080      3987  \n",
       "False Detection Errors          227          306      1333  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "inhouse_raw = pd.DataFrame(reversed([\n",
    "    [12929, 438, 1200, int(inhouse_1[\"arabida\"] * 2.087), int(inhouse_5[\"arabida\"] * 2.087),\n",
    "     int(inhouse_mq[\"arabida\"] * 2.087)], #False Detection Errors\n",
    "    [1940, 174, 1357, inhouse_1['foreign'], inhouse_5['foreign'],\n",
    "     int(inhouse_mq[\"foreign\"])], # Foreign Peak Errors\n",
    "    [1239, 207, 1244, 5793, 6610, 0]  # Native Peak Errors\n",
    "]),\n",
    "columns = ['FlashLFQ v1', 'FlashLFQ v2 1%', 'FlashLFQ v2 5%',  'IonQuant 1%', 'IonQuant 5%', \"MaxQuant\"],\n",
    "index = reversed([\"False Detection Errors\", \"Foreign Peak Errors\", \"Native Peak Errors\"]))\n",
    "inhouse_raw"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FlashLFQ v1</th>\n",
       "      <th>FlashLFQ v2 1%</th>\n",
       "      <th>FlashLFQ v2 5%</th>\n",
       "      <th>IonQuant 1%</th>\n",
       "      <th>IonQuant 5%</th>\n",
       "      <th>MaxQuant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Native Peak Errors</th>\n",
       "      <td>0.692</td>\n",
       "      <td>0.295</td>\n",
       "      <td>0.915</td>\n",
       "      <td>5.440</td>\n",
       "      <td>4.640</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Foreign Peak Errors</th>\n",
       "      <td>1.036</td>\n",
       "      <td>0.246</td>\n",
       "      <td>0.984</td>\n",
       "      <td>0.358</td>\n",
       "      <td>0.826</td>\n",
       "      <td>1.627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>False Detection Errors</th>\n",
       "      <td>6.906</td>\n",
       "      <td>0.621</td>\n",
       "      <td>0.871</td>\n",
       "      <td>0.212</td>\n",
       "      <td>0.235</td>\n",
       "      <td>0.544</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        FlashLFQ v1  FlashLFQ v2 1%  FlashLFQ v2 5%  \\\n",
       "Native Peak Errors            0.692           0.295           0.915   \n",
       "Foreign Peak Errors           1.036           0.246           0.984   \n",
       "False Detection Errors        6.906           0.621           0.871   \n",
       "\n",
       "                        IonQuant 1%  IonQuant 5%  MaxQuant  \n",
       "Native Peak Errors            5.440        4.640     0.000  \n",
       "Foreign Peak Errors           0.358        0.826     1.627  \n",
       "False Detection Errors        0.212        0.235     0.544  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "inhouse_pct = pd.DataFrame(reversed([\n",
    "    [6.906, 0.621, 0.871, round(100*(inhouse_1[\"arabida\"] * 2.087) / inhouse_1[\"total\"], 3), round(100*(inhouse_5[\"arabida\"] * 2.087) / inhouse_5[\"total\"], 3),\n",
    "     round(100* inhouse_mq[\"arabida\"] * 2.087 / inhouse_mq[\"total\"] ,3)], # False Detection\n",
    "    [1.036, 0.246, 0.984, round(100*inhouse_1['foreign']/inhouse_1[\"total\"], 3), round(100*inhouse_5['foreign']/inhouse_5[\"total\"], 3),\n",
    "     round(100* inhouse_mq[\"foreign\"]/ inhouse_mq[\"total\"] ,3)], # Foreign Peak\n",
    "    [0.692, 0.295, 0.915, 5.44, 4.64, 0],  # Native Peak\n",
    "]),\n",
    "columns = ['FlashLFQ v1', 'FlashLFQ v2 1%', 'FlashLFQ v2 5%',  'IonQuant 1%', 'IonQuant 5%', \"MaxQuant\"],\n",
    "index = reversed([\"False Detection Errors\", \"Foreign Peak Errors\", \"Native Peak Errors\"]))\n",
    "inhouse_pct"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def prep_df(df, name):\n",
    "    df = df.stack().reset_index()\n",
    "    df.columns = ['Error Type', 'engine', 'values']\n",
    "    df['DF'] = name\n",
    "    return df\n",
    "\n",
    "df1 = prep_df(gygi_raw, 'Lim et al.')\n",
    "df2 = prep_df(inhouse_raw, 'E. coli')\n",
    "df3 = prep_df(kelly_raw, 'Single Cell')\n",
    "\n",
    "df = pd.concat([df1, df2, df3])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.sort_values(by = \"Error Type\", inplace=True)\n",
    "df[\"Color\"] = df[\"engine\"] + df[\"Error Type\"]\n",
    "\n",
    "color = []\n",
    "for x in df[\"Color\"].tolist():\n",
    "    if (\"Foreign\" in x):\n",
    "        if('FlashLFQ v2 1%' in x):\n",
    "            color.append('FlashLFQ v2 1%')\n",
    "        elif('FlashLFQ v2 5%' in x):\n",
    "            color.append('FlashLFQ v2 5%')\n",
    "        elif(\"IonQuant 1%\" in x):\n",
    "            color.append('IonQuant 1%')\n",
    "        elif(\"IonQuant 5%\" in x):\n",
    "            color.append('IonQuant 5%')\n",
    "        elif('FlashLFQ v1' in x):\n",
    "            color.append('FlashLFQ v1')\n",
    "        elif('MaxQuant' in x):\n",
    "            color.append('MaxQuant')\n",
    "        else:\n",
    "            color.append(x[0:8])\n",
    "    else:\n",
    "        color.append(x)\n",
    "\n",
    "df[\"Software\"] = color"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<style>\n",
       "  #altair-viz-5d24477307be42a3989fc8c480a72441.vega-embed {\n",
       "    width: 100%;\n",
       "    display: flex;\n",
       "  }\n",
       "\n",
       "  #altair-viz-5d24477307be42a3989fc8c480a72441.vega-embed details,\n",
       "  #altair-viz-5d24477307be42a3989fc8c480a72441.vega-embed details summary {\n",
       "    position: relative;\n",
       "  }\n",
       "</style>\n",
       "<div id=\"altair-viz-5d24477307be42a3989fc8c480a72441\"></div>\n",
       "<script type=\"text/javascript\">\n",
       "  var VEGA_DEBUG = (typeof VEGA_DEBUG == \"undefined\") ? {} : VEGA_DEBUG;\n",
       "  (function(spec, embedOpt){\n",
       "    let outputDiv = document.currentScript.previousElementSibling;\n",
       "    if (outputDiv.id !== \"altair-viz-5d24477307be42a3989fc8c480a72441\") {\n",
       "      outputDiv = document.getElementById(\"altair-viz-5d24477307be42a3989fc8c480a72441\");\n",
       "    }\n",
       "    const paths = {\n",
       "      \"vega\": \"https://cdn.jsdelivr.net/npm/vega@5?noext\",\n",
       "      \"vega-lib\": \"https://cdn.jsdelivr.net/npm/vega-lib?noext\",\n",
       "      \"vega-lite\": \"https://cdn.jsdelivr.net/npm/vega-lite@5.16.3?noext\",\n",
       "      \"vega-embed\": \"https://cdn.jsdelivr.net/npm/vega-embed@6?noext\",\n",
       "    };\n",
       "\n",
       "    function maybeLoadScript(lib, version) {\n",
       "      var key = `${lib.replace(\"-\", \"\")}_version`;\n",
       "      return (VEGA_DEBUG[key] == version) ?\n",
       "        Promise.resolve(paths[lib]) :\n",
       "        new Promise(function(resolve, reject) {\n",
       "          var s = document.createElement('script');\n",
       "          document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "          s.async = true;\n",
       "          s.onload = () => {\n",
       "            VEGA_DEBUG[key] = version;\n",
       "            return resolve(paths[lib]);\n",
       "          };\n",
       "          s.onerror = () => reject(`Error loading script: ${paths[lib]}`);\n",
       "          s.src = paths[lib];\n",
       "        });\n",
       "    }\n",
       "\n",
       "    function showError(err) {\n",
       "      outputDiv.innerHTML = `<div class=\"error\" style=\"color:red;\">${err}</div>`;\n",
       "      throw err;\n",
       "    }\n",
       "\n",
       "    function displayChart(vegaEmbed) {\n",
       "      vegaEmbed(outputDiv, spec, embedOpt)\n",
       "        .catch(err => showError(`Javascript Error: ${err.message}<br>This usually means there's a typo in your chart specification. See the javascript console for the full traceback.`));\n",
       "    }\n",
       "\n",
       "    if(typeof define === \"function\" && define.amd) {\n",
       "      requirejs.config({paths});\n",
       "      require([\"vega-embed\"], displayChart, err => showError(`Error loading script: ${err.message}`));\n",
       "    } else {\n",
       "      maybeLoadScript(\"vega\", \"5\")\n",
       "        .then(() => maybeLoadScript(\"vega-lite\", \"5.16.3\"))\n",
       "        .then(() => maybeLoadScript(\"vega-embed\", \"6\"))\n",
       "        .catch(showError)\n",
       "        .then(() => displayChart(vegaEmbed));\n",
       "    }\n",
       "  })({\"config\": {\"view\": {\"continuousWidth\": 300, \"continuousHeight\": 300, \"strokeOpacity\": 0}, \"axis\": {\"labelFontSize\": 18, \"titleFontSize\": 20}, \"legend\": {\"labelFontSize\": 18, \"titleFontSize\": 22}}, \"data\": {\"name\": \"data-1bb5fcd5e032a1e153268515758032e7\"}, \"mark\": {\"type\": \"bar\"}, \"encoding\": {\"color\": {\"field\": \"Software\", \"legend\": {\"values\": [\"FlashLFQ v1\", \"FlashLFQ v2 1%\", \"FlashLFQ v2 5%\", \"IonQuant 1%\", \"IonQuant 5%\", \"MaxQuant\"]}, \"scale\": {\"range\": [\"#13459c\", \"#2382f7\", \"#a4cefc\", \"#118C53\", \"#47B384\", \"#B8E6DF\", \"#118c11\", \"#47b347\", \"#b8e6b8\", \"#9f040e\", \"#e30613\", \"#fc9ca2\", \"#ff7b00\", \"#ffaa00\", \"#ffe15c\", \"#5A108F\", \"#AB51E3\", \"#DC97FF\"]}, \"sort\": [\"FlashLFQ v1\", \"FlashLFQ v1False Detection Errors\", \"FlashLFQ v1Native Peak Errors\", \"FlashLFQ v2 1%\", \"FlashLFQ v2 1%False Detection Errors\", \"FlashLFQ v2 1%Native Peak Errors\", \"FlashLFQ v2 5%\", \"FlashLFQ v2 5%False Detection Errors\", \"FlashLFQ v2 5%Native Peak Errors\", \"IonQuant 1%\", \"IonQuant 1%False Detection Errors\", \"IonQuant 1%Native Peak Errors\", \"IonQuant 5%\", \"IonQuant 5%False Detection Errors\", \"IonQuant 5%Native Peak Errors\", \"MaxQuant\", \"MaxQuantFalse Detection Errors\", \"MaxQuantNative Peak Errors\"], \"type\": \"nominal\"}, \"column\": {\"field\": \"DF\", \"header\": {\"labelFontSize\": 24}, \"spacing\": 45, \"title\": null, \"type\": \"nominal\"}, \"order\": {\"field\": \"Software\", \"sort\": \"ascending\", \"type\": \"nominal\"}, \"x\": {\"axis\": {\"labelAngle\": -45}, \"field\": \"engine\", \"sort\": [\"FlashLFQ v1\", \"FlashLFQ v2 1%\", \"FlashLFQ v2 5%\", \"IonQuant 1%\", \"IonQuant 5%\", \"MaxQuant\"], \"title\": null, \"type\": \"nominal\"}, \"y\": {\"aggregate\": \"sum\", \"axis\": {\"grid\": false, \"title\": \"Number of False Detections\"}, \"field\": \"values\", \"type\": \"quantitative\"}}, \"height\": 350, \"width\": 200, \"$schema\": \"https://vega.github.io/schema/vega-lite/v5.16.3.json\", \"datasets\": {\"data-1bb5fcd5e032a1e153268515758032e7\": [{\"Error Type\": \"False Detection Errors\", \"engine\": \"MaxQuant\", \"values\": 116, \"DF\": \"Single Cell\", \"Color\": \"MaxQuantFalse Detection Errors\", \"Software\": \"MaxQuantFalse Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 12929, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v1False Detection Errors\", \"Software\": \"FlashLFQ v1False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v2 1%\", \"values\": 438, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v2 1%False Detection Errors\", \"Software\": \"FlashLFQ v2 1%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 1200, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v2 5%False Detection Errors\", \"Software\": \"FlashLFQ v2 5%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"IonQuant 1%\", \"values\": 227, \"DF\": \"E. coli\", \"Color\": \"IonQuant 1%False Detection Errors\", \"Software\": \"IonQuant 1%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"IonQuant 5%\", \"values\": 306, \"DF\": \"E. coli\", \"Color\": \"IonQuant 5%False Detection Errors\", \"Software\": \"IonQuant 5%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"MaxQuant\", \"values\": 1270, \"DF\": \"Lim et al.\", \"Color\": \"MaxQuantFalse Detection Errors\", \"Software\": \"MaxQuantFalse Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"IonQuant 5%\", \"values\": 428, \"DF\": \"Lim et al.\", \"Color\": \"IonQuant 5%False Detection Errors\", \"Software\": \"IonQuant 5%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"IonQuant 1%\", \"values\": 234, \"DF\": \"Lim et al.\", \"Color\": \"IonQuant 1%False Detection Errors\", \"Software\": \"IonQuant 1%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 1680, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v2 5%False Detection Errors\", \"Software\": \"FlashLFQ v2 5%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 15992, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v1False Detection Errors\", \"Software\": \"FlashLFQ v1False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v2 1%\", \"values\": 572, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v2 1%False Detection Errors\", \"Software\": \"FlashLFQ v2 1%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 1613, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v1False Detection Errors\", \"Software\": \"FlashLFQ v1False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v2 1%\", \"values\": 6, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v2 1%False Detection Errors\", \"Software\": \"FlashLFQ v2 1%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 198, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v2 5%False Detection Errors\", \"Software\": \"FlashLFQ v2 5%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"IonQuant 1%\", \"values\": 2, \"DF\": \"Single Cell\", \"Color\": \"IonQuant 1%False Detection Errors\", \"Software\": \"IonQuant 1%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"MaxQuant\", \"values\": 1333, \"DF\": \"E. coli\", \"Color\": \"MaxQuantFalse Detection Errors\", \"Software\": \"MaxQuantFalse Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"IonQuant 5%\", \"values\": 48, \"DF\": \"Single Cell\", \"Color\": \"IonQuant 5%False Detection Errors\", \"Software\": \"IonQuant 5%False Detection Errors\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 3558, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v1Foreign Peak Errors\", \"Software\": \"FlashLFQ v1\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"IonQuant 1%\", \"values\": 383, \"DF\": \"E. coli\", \"Color\": \"IonQuant 1%Foreign Peak Errors\", \"Software\": \"IonQuant 1%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v2 1%\", \"values\": 7, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v2 1%Foreign Peak Errors\", \"Software\": \"FlashLFQ v2 1%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 1193, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v2 5%Foreign Peak Errors\", \"Software\": \"FlashLFQ v2 5%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"IonQuant 1%\", \"values\": 66, \"DF\": \"Single Cell\", \"Color\": \"IonQuant 1%Foreign Peak Errors\", \"Software\": \"IonQuant 1%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"IonQuant 5%\", \"values\": 1124, \"DF\": \"Single Cell\", \"Color\": \"IonQuant 5%Foreign Peak Errors\", \"Software\": \"IonQuant 5%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"MaxQuant\", \"values\": 1127, \"DF\": \"Single Cell\", \"Color\": \"MaxQuantForeign Peak Errors\", \"Software\": \"MaxQuant\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"MaxQuant\", \"values\": 3987, \"DF\": \"E. coli\", \"Color\": \"MaxQuantForeign Peak Errors\", \"Software\": \"MaxQuant\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"IonQuant 5%\", \"values\": 1080, \"DF\": \"E. coli\", \"Color\": \"IonQuant 5%Foreign Peak Errors\", \"Software\": \"IonQuant 5%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 1357, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v2 5%Foreign Peak Errors\", \"Software\": \"FlashLFQ v2 5%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 1940, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v1Foreign Peak Errors\", \"Software\": \"FlashLFQ v1\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 4870, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v1Foreign Peak Errors\", \"Software\": \"FlashLFQ v1\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"MaxQuant\", \"values\": 21831, \"DF\": \"Lim et al.\", \"Color\": \"MaxQuantForeign Peak Errors\", \"Software\": \"MaxQuant\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v2 1%\", \"values\": 407, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v2 1%Foreign Peak Errors\", \"Software\": \"FlashLFQ v2 1%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 4535, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v2 5%Foreign Peak Errors\", \"Software\": \"FlashLFQ v2 5%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"IonQuant 1%\", \"values\": 1498, \"DF\": \"Lim et al.\", \"Color\": \"IonQuant 1%Foreign Peak Errors\", \"Software\": \"IonQuant 1%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"IonQuant 5%\", \"values\": 6154, \"DF\": \"Lim et al.\", \"Color\": \"IonQuant 5%Foreign Peak Errors\", \"Software\": \"IonQuant 5%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v2 1%\", \"values\": 174, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v2 1%Foreign Peak Errors\", \"Software\": \"FlashLFQ v2 1%\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v2 1%\", \"values\": 230, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v2 1%Native Peak Errors\", \"Software\": \"FlashLFQ v2 1%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 1003, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v2 5%Native Peak Errors\", \"Software\": \"FlashLFQ v2 5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"IonQuant 1%\", \"values\": 18118, \"DF\": \"Lim et al.\", \"Color\": \"IonQuant 1%Native Peak Errors\", \"Software\": \"IonQuant 1%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"IonQuant 5%\", \"values\": 19649, \"DF\": \"Lim et al.\", \"Color\": \"IonQuant 5%Native Peak Errors\", \"Software\": \"IonQuant 5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"MaxQuant\", \"values\": 0, \"DF\": \"Lim et al.\", \"Color\": \"MaxQuantNative Peak Errors\", \"Software\": \"MaxQuantNative Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"IonQuant 5%\", \"values\": 386, \"DF\": \"Single Cell\", \"Color\": \"IonQuant 5%Native Peak Errors\", \"Software\": \"IonQuant 5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"IonQuant 1%\", \"values\": 26, \"DF\": \"Single Cell\", \"Color\": \"IonQuant 1%Native Peak Errors\", \"Software\": \"IonQuant 1%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 325, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v2 5%Native Peak Errors\", \"Software\": \"FlashLFQ v2 5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v2 1%\", \"values\": 0, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v2 1%Native Peak Errors\", \"Software\": \"FlashLFQ v2 1%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 228, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v1Native Peak Errors\", \"Software\": \"FlashLFQ v1Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 1239, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v1Native Peak Errors\", \"Software\": \"FlashLFQ v1Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v2 1%\", \"values\": 207, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v2 1%Native Peak Errors\", \"Software\": \"FlashLFQ v2 1%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 1244, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v2 5%Native Peak Errors\", \"Software\": \"FlashLFQ v2 5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"IonQuant 1%\", \"values\": 5793, \"DF\": \"E. coli\", \"Color\": \"IonQuant 1%Native Peak Errors\", \"Software\": \"IonQuant 1%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"IonQuant 5%\", \"values\": 6610, \"DF\": \"E. coli\", \"Color\": \"IonQuant 5%Native Peak Errors\", \"Software\": \"IonQuant 5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"MaxQuant\", \"values\": 0, \"DF\": \"E. coli\", \"Color\": \"MaxQuantNative Peak Errors\", \"Software\": \"MaxQuantNative Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"MaxQuant\", \"values\": 0, \"DF\": \"Single Cell\", \"Color\": \"MaxQuantNative Peak Errors\", \"Software\": \"MaxQuantNative Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 1040, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v1Native Peak Errors\", \"Software\": \"FlashLFQ v1Native Peak Errors\"}]}}, {\"mode\": \"vega-lite\"});\n",
       "</script>"
      ],
      "text/plain": [
       "alt.Chart(...)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "label_order = [ \n",
    " 'FlashLFQ v1', 'FlashLFQ v1False Detection Errors', 'FlashLFQ v1Native Peak Errors',\n",
    " 'FlashLFQ v2 1%', 'FlashLFQ v2 1%False Detection Errors', 'FlashLFQ v2 1%Native Peak Errors',\n",
    " 'FlashLFQ v2 5%', 'FlashLFQ v2 5%False Detection Errors', 'FlashLFQ v2 5%Native Peak Errors',\n",
    "  'IonQuant 1%', 'IonQuant 1%False Detection Errors', 'IonQuant 1%Native Peak Errors',\n",
    "  'IonQuant 5%', 'IonQuant 5%False Detection Errors', 'IonQuant 5%Native Peak Errors',\n",
    "  'MaxQuant', 'MaxQuantFalse Detection Errors', 'MaxQuantNative Peak Errors'\n",
    "  ]\n",
    "\n",
    "alt.Chart(df).mark_bar().encode(\n",
    "\n",
    "    # tell Altair which field to group columns on\n",
    "    x=alt.X('engine:N', title=None, sort = [\"FlashLFQ v1\", 'FlashLFQ v2 1%', 'FlashLFQ v2 5%', \"IonQuant 1%\", \"IonQuant 5%\", \"MaxQuant\"], axis=alt.Axis(labelAngle=-45)),\n",
    "\n",
    "    # tell Altair which field to use as Y values and how to calculate\n",
    "    y=alt.Y('sum(values):Q', \n",
    "        axis=alt.Axis(\n",
    "            grid=False,\n",
    "            title=\"Number of False Detections\")),\n",
    "\n",
    "    # tell Altair which field to use to use as the set of columns to be  represented in each group\n",
    "    column=alt.Column('DF:N', title=None, header=alt.Header(labelFontSize=24), spacing=45),\n",
    "\n",
    "    # tell Altair which field to use for color segmentation \n",
    "    color=alt.Color('Software:N', sort = label_order,\n",
    "            scale=alt.Scale(\n",
    "                # make it look pretty with an enjoyable color pallet\n",
    "                range=['#13459c', '#2382f7','#a4cefc',\n",
    "                       '#118C53', '#47B384','#B8E6DF',\n",
    "                       '#118c11', '#47b347','#b8e6b8',\n",
    "                       '#9f040e', '#e30613','#fc9ca2',\n",
    "                       '#ff7b00', '#ffaa00','#ffe15c',\n",
    "                       '#5A108F', '#AB51E3','#DC97FF'\n",
    "                       ],\n",
    "            ),\n",
    "            legend=alt.Legend(values=[\"FlashLFQ v1\", \"FlashLFQ v2 1%\", \"FlashLFQ v2 5%\", \"IonQuant 1%\", \"IonQuant 5%\", \"MaxQuant\"])\n",
    "        ),\n",
    "     \n",
    "    order = alt.Order(\"Software:N\", sort = 'ascending')\n",
    "        )\\\n",
    "    .configure_view(\n",
    "        # remove grid lines around column clusters\n",
    "        strokeOpacity=0   \n",
    "    )\\\n",
    "        .properties(width = 200, height = 350)\\\n",
    "        .configure_legend(titleFontSize = 22, labelFontSize =18)\\\n",
    "        .configure_axis(titleFontSize = 20, labelFontSize =18)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def prep_df(df, name):\n",
    "    df = df.stack().reset_index()\n",
    "    df.columns = ['Error Type', 'engine', 'values']\n",
    "    df['DF'] = name\n",
    "    return df\n",
    "\n",
    "df1 = prep_df(gygi_pct, 'Lim et al.')\n",
    "df2 = prep_df(inhouse_pct, 'E. coli')\n",
    "df3 = prep_df(kelly_pct, 'Single Cell')\n",
    "\n",
    "df = pd.concat([df1, df2, df3])\n",
    "\n",
    "df.sort_values(by = \"Error Type\", inplace=True)\n",
    "df[\"Color\"] = df[\"engine\"] + df[\"Error Type\"]\n",
    "\n",
    "color = []\n",
    "for x in df[\"Color\"].tolist():\n",
    "    if (\"Foreign\" in x):\n",
    "        if('FlashLFQ v2 2.5%' in x):\n",
    "            color.append('FlashLFQ v2 2.5%')\n",
    "        elif('FlashLFQ v2 5%' in x):\n",
    "            color.append('FlashLFQ v2 5%')\n",
    "        elif(\"IonQuant 1%\" in x):\n",
    "            color.append('IonQuant 1%')\n",
    "        elif(\"IonQuant 5%\" in x):\n",
    "            color.append('IonQuant 5%')\n",
    "        elif('FlashLFQ v1' in x):\n",
    "            color.append('FlashLFQ v1')\n",
    "        else:\n",
    "            color.append(x[0:8])\n",
    "    else:\n",
    "        color.append(x)\n",
    "\n",
    "df[\"Software\"] = color"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<style>\n",
       "  #altair-viz-8749615a353740d8a092a7320e1fb99b.vega-embed {\n",
       "    width: 100%;\n",
       "    display: flex;\n",
       "  }\n",
       "\n",
       "  #altair-viz-8749615a353740d8a092a7320e1fb99b.vega-embed details,\n",
       "  #altair-viz-8749615a353740d8a092a7320e1fb99b.vega-embed details summary {\n",
       "    position: relative;\n",
       "  }\n",
       "</style>\n",
       "<div id=\"altair-viz-8749615a353740d8a092a7320e1fb99b\"></div>\n",
       "<script type=\"text/javascript\">\n",
       "  var VEGA_DEBUG = (typeof VEGA_DEBUG == \"undefined\") ? {} : VEGA_DEBUG;\n",
       "  (function(spec, embedOpt){\n",
       "    let outputDiv = document.currentScript.previousElementSibling;\n",
       "    if (outputDiv.id !== \"altair-viz-8749615a353740d8a092a7320e1fb99b\") {\n",
       "      outputDiv = document.getElementById(\"altair-viz-8749615a353740d8a092a7320e1fb99b\");\n",
       "    }\n",
       "    const paths = {\n",
       "      \"vega\": \"https://cdn.jsdelivr.net/npm/vega@5?noext\",\n",
       "      \"vega-lib\": \"https://cdn.jsdelivr.net/npm/vega-lib?noext\",\n",
       "      \"vega-lite\": \"https://cdn.jsdelivr.net/npm/vega-lite@5.16.3?noext\",\n",
       "      \"vega-embed\": \"https://cdn.jsdelivr.net/npm/vega-embed@6?noext\",\n",
       "    };\n",
       "\n",
       "    function maybeLoadScript(lib, version) {\n",
       "      var key = `${lib.replace(\"-\", \"\")}_version`;\n",
       "      return (VEGA_DEBUG[key] == version) ?\n",
       "        Promise.resolve(paths[lib]) :\n",
       "        new Promise(function(resolve, reject) {\n",
       "          var s = document.createElement('script');\n",
       "          document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "          s.async = true;\n",
       "          s.onload = () => {\n",
       "            VEGA_DEBUG[key] = version;\n",
       "            return resolve(paths[lib]);\n",
       "          };\n",
       "          s.onerror = () => reject(`Error loading script: ${paths[lib]}`);\n",
       "          s.src = paths[lib];\n",
       "        });\n",
       "    }\n",
       "\n",
       "    function showError(err) {\n",
       "      outputDiv.innerHTML = `<div class=\"error\" style=\"color:red;\">${err}</div>`;\n",
       "      throw err;\n",
       "    }\n",
       "\n",
       "    function displayChart(vegaEmbed) {\n",
       "      vegaEmbed(outputDiv, spec, embedOpt)\n",
       "        .catch(err => showError(`Javascript Error: ${err.message}<br>This usually means there's a typo in your chart specification. See the javascript console for the full traceback.`));\n",
       "    }\n",
       "\n",
       "    if(typeof define === \"function\" && define.amd) {\n",
       "      requirejs.config({paths});\n",
       "      require([\"vega-embed\"], displayChart, err => showError(`Error loading script: ${err.message}`));\n",
       "    } else {\n",
       "      maybeLoadScript(\"vega\", \"5\")\n",
       "        .then(() => maybeLoadScript(\"vega-lite\", \"5.16.3\"))\n",
       "        .then(() => maybeLoadScript(\"vega-embed\", \"6\"))\n",
       "        .catch(showError)\n",
       "        .then(() => displayChart(vegaEmbed));\n",
       "    }\n",
       "  })({\"config\": {\"view\": {\"continuousWidth\": 300, \"continuousHeight\": 300, \"strokeOpacity\": 0}, \"axis\": {\"labelFontSize\": 24, \"titleFontSize\": 32}, \"legend\": {\"labelFontSize\": 18, \"titleFontSize\": 32}}, \"data\": {\"name\": \"data-66bf924f198f253188730e8a343e5778\"}, \"mark\": {\"type\": \"bar\"}, \"encoding\": {\"color\": {\"field\": \"Software\", \"legend\": {\"values\": [\"FlashLFQ v1\", \"FlashLFQ v2 2.5%\", \"FlashLFQ v2 5%\", \"IonQuant 1%\", \"IonQuant 5%\", \"MaxQuant\"]}, \"scale\": {\"range\": [\"#13459c\", \"#2382f7\", \"#a4cefc\", \"#118C53\", \"#47B384\", \"#B8E6DF\", \"#118c11\", \"#47b347\", \"#b8e6b8\", \"#E54501\", \"#F48C06\", \"#FAB82A\", \"#9f040e\", \"#CC3943\", \"#EA7981\", \"#44485D\", \"#8D99AE\", \"#A5B0C0\"]}, \"sort\": [\"FlashLFQ v1\", \"FlashLFQ v1False Detection Errors\", \"FlashLFQ v1Native Peak Errors\", \"FlashLFQ v2 2.5%\", \"FlashLFQ v2 2.5%False Detection Errors\", \"FlashLFQ v2 2.5%Native Peak Errors\", \"FlashLFQ v2 5%\", \"FlashLFQ v2 5%False Detection Errors\", \"FlashLFQ v2 5%Native Peak Errors\", \"IonQuant 1%\", \"IonQuant 1%False Detection Errors\", \"IonQuant 1%Native Peak Errors\", \"IonQuant 5%\", \"IonQuant 5%False Detection Errors\", \"IonQuant 5%Native Peak Errors\", \"MaxQuant\", \"MaxQuantFalse Detection Errors\", \"MaxQuantNative Peak Errors\"], \"type\": \"nominal\"}, \"column\": {\"field\": \"DF\", \"header\": {\"labelFontSize\": 24}, \"spacing\": 45, \"title\": null, \"type\": \"nominal\"}, \"order\": {\"field\": \"Software\", \"sort\": \"ascending\", \"type\": \"nominal\"}, \"x\": {\"axis\": {\"labelAngle\": -45}, \"field\": \"engine\", \"sort\": [\"FlashLFQ v1\", \"FlashLFQ v2 2.5%\", \"FlashLFQ v2 5%\", \"IonQuant 1%\", \"IonQuant 5%\", \"MaxQuant\"], \"title\": null, \"type\": \"nominal\"}, \"y\": {\"aggregate\": \"sum\", \"axis\": {\"grid\": false, \"title\": \"Percent False Detections\"}, \"field\": \"values\", \"type\": \"quantitative\"}}, \"height\": 700, \"width\": 400, \"$schema\": \"https://vega.github.io/schema/vega-lite/v5.16.3.json\", \"datasets\": {\"data-66bf924f198f253188730e8a343e5778\": [{\"Error Type\": \"False Detection Errors\", \"engine\": \"MaxQuant\", \"values\": 0.4863639079772931, \"DF\": \"Single Cell\", \"Color\": \"MaxQuantFalse Detection Errors\", \"Software\": \"MaxQuantFalse Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 3.89407099949017, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v1False Detection Errors\", \"Software\": \"FlashLFQ v1False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v2 2.5%\", \"values\": 0.5266067301371896, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v2 2.5%False Detection Errors\", \"Software\": \"FlashLFQ v2 2.5%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 0.46370355027914556, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v2 5%False Detection Errors\", \"Software\": \"FlashLFQ v2 5%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"IonQuant 1%\", \"values\": 0.2394755254540875, \"DF\": \"E. coli\", \"Color\": \"IonQuant 1%False Detection Errors\", \"Software\": \"IonQuant 1%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"IonQuant 5%\", \"values\": 0.27085894020017254, \"DF\": \"E. coli\", \"Color\": \"IonQuant 5%False Detection Errors\", \"Software\": \"IonQuant 5%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"MaxQuant\", \"values\": 0.2534622511782428, \"DF\": \"Lim et al.\", \"Color\": \"MaxQuantFalse Detection Errors\", \"Software\": \"MaxQuantFalse Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"IonQuant 5%\", \"values\": 0.11264997978888462, \"DF\": \"Lim et al.\", \"Color\": \"IonQuant 5%False Detection Errors\", \"Software\": \"IonQuant 5%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"IonQuant 1%\", \"values\": 0.06866072652501715, \"DF\": \"Lim et al.\", \"Color\": \"IonQuant 1%False Detection Errors\", \"Software\": \"IonQuant 1%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 0.5734183455847732, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v2 5%False Detection Errors\", \"Software\": \"FlashLFQ v2 5%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 3.1566035550085076, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v1False Detection Errors\", \"Software\": \"FlashLFQ v1False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v2 2.5%\", \"values\": 0.42760173269674656, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v2 2.5%False Detection Errors\", \"Software\": \"FlashLFQ v2 2.5%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 1.1606534584636776, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v1False Detection Errors\", \"Software\": \"FlashLFQ v1False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v2 2.5%\", \"values\": 0.0, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v2 2.5%False Detection Errors\", \"Software\": \"FlashLFQ v2 2.5%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 0.25601378585197737, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v2 5%False Detection Errors\", \"Software\": \"FlashLFQ v2 5%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"IonQuant 1%\", \"values\": 0.06370573870573872, \"DF\": \"Single Cell\", \"Color\": \"IonQuant 1%False Detection Errors\", \"Software\": \"IonQuant 1%False Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"MaxQuant\", \"values\": 0.5427668456579191, \"DF\": \"E. coli\", \"Color\": \"MaxQuantFalse Detection Errors\", \"Software\": \"MaxQuantFalse Detection Errors\"}, {\"Error Type\": \"False Detection Errors\", \"engine\": \"IonQuant 5%\", \"values\": 0.08794269187304872, \"DF\": \"Single Cell\", \"Color\": \"IonQuant 5%False Detection Errors\", \"Software\": \"IonQuant 5%False Detection Errors\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 4.509584314937113, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v1Foreign Peak Errors\", \"Software\": \"FlashLFQ v1\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"IonQuant 1%\", \"values\": 0.37315869506404337, \"DF\": \"E. coli\", \"Color\": \"IonQuant 1%Foreign Peak Errors\", \"Software\": \"IonQuant 1%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v2 2.5%\", \"values\": 2.255639097744361, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v2 2.5%Foreign Peak Errors\", \"Software\": \"FlashLFQ v2 2.5%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 2.108768035516093, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v2 5%Foreign Peak Errors\", \"Software\": \"FlashLFQ v2 5%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"IonQuant 1%\", \"values\": 2.1367521367521367, \"DF\": \"Single Cell\", \"Color\": \"IonQuant 1%Foreign Peak Errors\", \"Software\": \"IonQuant 1%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"IonQuant 5%\", \"values\": 2.1873623321649522, \"DF\": \"Single Cell\", \"Color\": \"IonQuant 5%Foreign Peak Errors\", \"Software\": \"IonQuant 5%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"MaxQuant\", \"values\": 1.8942336420675232, \"DF\": \"Single Cell\", \"Color\": \"MaxQuantForeign Peak Errors\", \"Software\": \"MaxQuant\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"MaxQuant\", \"values\": 0.41661757031999597, \"DF\": \"E. coli\", \"Color\": \"MaxQuantForeign Peak Errors\", \"Software\": \"MaxQuant\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"IonQuant 5%\", \"values\": 0.8542832494827731, \"DF\": \"E. coli\", \"Color\": \"IonQuant 5%Foreign Peak Errors\", \"Software\": \"IonQuant 5%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 0.8802734025626783, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v2 5%Foreign Peak Errors\", \"Software\": \"FlashLFQ v2 5%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 1.1764901469642897, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v1Foreign Peak Errors\", \"Software\": \"FlashLFQ v1\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 1.9784237772870434, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v1Foreign Peak Errors\", \"Software\": \"FlashLFQ v1\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"MaxQuant\", \"values\": 0.37766451324764055, \"DF\": \"Lim et al.\", \"Color\": \"MaxQuantForeign Peak Errors\", \"Software\": \"MaxQuant\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v2 2.5%\", \"values\": 0.9561450659246846, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v2 2.5%Foreign Peak Errors\", \"Software\": \"FlashLFQ v2 2.5%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 1.8994809525656011, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v2 5%Foreign Peak Errors\", \"Software\": \"FlashLFQ v2 5%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"IonQuant 1%\", \"values\": 0.4361891706648389, \"DF\": \"Lim et al.\", \"Color\": \"IonQuant 1%Foreign Peak Errors\", \"Software\": \"IonQuant 1%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"IonQuant 5%\", \"values\": 1.6050288159675072, \"DF\": \"Lim et al.\", \"Color\": \"IonQuant 5%Foreign Peak Errors\", \"Software\": \"IonQuant 5%\"}, {\"Error Type\": \"Foreign Peak Errors\", \"engine\": \"FlashLFQ v2 2.5%\", \"values\": 0.5661457560272258, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v2 2.5%Foreign Peak Errors\", \"Software\": \"FlashLFQ v2 2.5%\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v2 2.5%\", \"values\": 0.3466464495842912, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v2 2.5%Native Peak Errors\", \"Software\": \"FlashLFQ v2 2.5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 0.4230760593114241, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v2 5%Native Peak Errors\", \"Software\": \"FlashLFQ v2 5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"IonQuant 1%\", \"values\": 0.7924279454462352, \"DF\": \"Lim et al.\", \"Color\": \"IonQuant 1%Native Peak Errors\", \"Software\": \"IonQuant 1%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"IonQuant 5%\", \"values\": 0.7759500427977474, \"DF\": \"Lim et al.\", \"Color\": \"IonQuant 5%Native Peak Errors\", \"Software\": \"IonQuant 5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"MaxQuant\", \"values\": 0.0, \"DF\": \"Lim et al.\", \"Color\": \"MaxQuantNative Peak Errors\", \"Software\": \"MaxQuantNative Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"IonQuant 5%\", \"values\": 1.0944211579920173, \"DF\": \"Single Cell\", \"Color\": \"IonQuant 5%Native Peak Errors\", \"Software\": \"IonQuant 5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"IonQuant 1%\", \"values\": 2.1603369869265823, \"DF\": \"Single Cell\", \"Color\": \"IonQuant 1%Native Peak Errors\", \"Software\": \"IonQuant 1%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 0.7248827525318735, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v2 5%Native Peak Errors\", \"Software\": \"FlashLFQ v2 5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v2 2.5%\", \"values\": 0.0, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v2 2.5%Native Peak Errors\", \"Software\": \"FlashLFQ v2 2.5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 0.3333951916500744, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v1Native Peak Errors\", \"Software\": \"FlashLFQ v1Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 0.8216749127520461, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v1Native Peak Errors\", \"Software\": \"FlashLFQ v1Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v2 2.5%\", \"values\": 0.7761322137874861, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v2 2.5%Native Peak Errors\", \"Software\": \"FlashLFQ v2 2.5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 0.8855781650162723, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v2 5%Native Peak Errors\", \"Software\": \"FlashLFQ v2 5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"IonQuant 1%\", \"values\": 0.3055160348891145, \"DF\": \"E. coli\", \"Color\": \"IonQuant 1%Native Peak Errors\", \"Software\": \"IonQuant 1%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"IonQuant 5%\", \"values\": 0.3540079294125693, \"DF\": \"E. coli\", \"Color\": \"IonQuant 5%Native Peak Errors\", \"Software\": \"IonQuant 5%Native Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"MaxQuant\", \"values\": 0.0, \"DF\": \"E. coli\", \"Color\": \"MaxQuantNative Peak Errors\", \"Software\": \"MaxQuantNative Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"MaxQuant\", \"values\": 0.0, \"DF\": \"Single Cell\", \"Color\": \"MaxQuantNative Peak Errors\", \"Software\": \"MaxQuantNative Peak Errors\"}, {\"Error Type\": \"Native Peak Errors\", \"engine\": \"FlashLFQ v1\", \"values\": 0.3400601428113048, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v1Native Peak Errors\", \"Software\": \"FlashLFQ v1Native Peak Errors\"}]}}, {\"mode\": \"vega-lite\"});\n",
       "</script>"
      ],
      "text/plain": [
       "alt.Chart(...)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "label_order = [ \n",
    " 'FlashLFQ v1', 'FlashLFQ v1False Detection Errors', 'FlashLFQ v1Native Peak Errors',\n",
    " 'FlashLFQ v2 2.5%', 'FlashLFQ v2 2.5%False Detection Errors', 'FlashLFQ v2 2.5%Native Peak Errors',\n",
    " 'FlashLFQ v2 5%', 'FlashLFQ v2 5%False Detection Errors', 'FlashLFQ v2 5%Native Peak Errors',\n",
    "  'IonQuant 1%', 'IonQuant 1%False Detection Errors', 'IonQuant 1%Native Peak Errors',\n",
    "  'IonQuant 5%', 'IonQuant 5%False Detection Errors', 'IonQuant 5%Native Peak Errors',\n",
    "  'MaxQuant', 'MaxQuantFalse Detection Errors', 'MaxQuantNative Peak Errors'\n",
    "  ]\n",
    "\n",
    "alt.Chart(df).mark_bar().encode(\n",
    "\n",
    "    # tell Altair which field to group columns on\n",
    "    x=alt.X('engine:N', title=None, sort = [\"FlashLFQ v1\", 'FlashLFQ v2 2.5%', 'FlashLFQ v2 5%', \"IonQuant 1%\", \"IonQuant 5%\", \"MaxQuant\"], axis=alt.Axis(labelAngle=-45)),\n",
    "\n",
    "    # tell Altair which field to use as Y values and how to calculate\n",
    "    y=alt.Y('sum(values):Q', \n",
    "        axis=alt.Axis(\n",
    "            grid=False,\n",
    "            title=\"Percent False Detections\")),\n",
    "\n",
    "    # tell Altair which field to use to use as the set of columns to be  represented in each group\n",
    "    column=alt.Column('DF:N', title=None, header=alt.Header(labelFontSize=24), spacing=45),\n",
    "\n",
    "    # tell Altair which field to use for color segmentation \n",
    "    color=alt.Color('Software:N', sort = label_order,\n",
    "            scale=alt.Scale(\n",
    "                # make it look pretty with an enjoyable color pallet\n",
    "                range=['#13459c', '#2382f7','#a4cefc',\n",
    "                       '#118C53', '#47B384','#B8E6DF',\n",
    "                       '#118c11', '#47b347','#b8e6b8',\n",
    "                       '#E54501', '#F48C06','#FAB82A',\n",
    "                       '#9f040e', '#CC3943','#EA7981',\n",
    "                       '#44485D', '#8D99AE','#A5B0C0'\n",
    "                       ],\n",
    "            ),\n",
    "            legend=alt.Legend(values=[\"FlashLFQ v1\", \"FlashLFQ v2 2.5%\", \"FlashLFQ v2 5%\", \"IonQuant 1%\", \"IonQuant 5%\", \"MaxQuant\"])\n",
    "        ),\n",
    "     \n",
    "    order = alt.Order(\"Software:N\", sort = 'ascending')\n",
    "        )\\\n",
    "    .configure_view(\n",
    "        # remove grid lines around column clusters\n",
    "        strokeOpacity=0   \n",
    "    )\\\n",
    "        .properties(width = 400, height = 700)\\\n",
    "        .configure_legend(titleFontSize = 32, labelFontSize =18)\\\n",
    "        .configure_axis(titleFontSize = 32, labelFontSize =24)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FlashLFQ v1</th>\n",
       "      <th>FlashLFQ v2 2.5%</th>\n",
       "      <th>FlashLFQ v2 5%</th>\n",
       "      <th>IonQuant 1%</th>\n",
       "      <th>IonQuant 5%</th>\n",
       "      <th>MaxQuant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PIP Identified Peak Traces</th>\n",
       "      <td>276230</td>\n",
       "      <td>158135</td>\n",
       "      <td>180523</td>\n",
       "      <td>364750</td>\n",
       "      <td>413139</td>\n",
       "      <td>255253</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            FlashLFQ v1  FlashLFQ v2 2.5%  FlashLFQ v2 5%  \\\n",
       "PIP Identified Peak Traces       276230            158135          180523   \n",
       "\n",
       "                            IonQuant 1%  IonQuant 5%  MaxQuant  \n",
       "PIP Identified Peak Traces       364750       413139    255253  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gygi_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>counts</th>\n",
       "      <th>engine</th>\n",
       "      <th>values</th>\n",
       "      <th>DF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v1</td>\n",
       "      <td>276230</td>\n",
       "      <td>Lim et al.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v2 2.5%</td>\n",
       "      <td>158135</td>\n",
       "      <td>Lim et al.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v2 5%</td>\n",
       "      <td>180523</td>\n",
       "      <td>Lim et al.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>IonQuant 1%</td>\n",
       "      <td>364750</td>\n",
       "      <td>Lim et al.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>IonQuant 5%</td>\n",
       "      <td>413139</td>\n",
       "      <td>Lim et al.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>MaxQuant</td>\n",
       "      <td>255253</td>\n",
       "      <td>Lim et al.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       counts            engine  values          DF\n",
       "0  PIP Identified Peak Traces       FlashLFQ v1  276230  Lim et al.\n",
       "1  PIP Identified Peak Traces  FlashLFQ v2 2.5%  158135  Lim et al.\n",
       "2  PIP Identified Peak Traces    FlashLFQ v2 5%  180523  Lim et al.\n",
       "3  PIP Identified Peak Traces       IonQuant 1%  364750  Lim et al.\n",
       "4  PIP Identified Peak Traces       IonQuant 5%  413139  Lim et al.\n",
       "5  PIP Identified Peak Traces          MaxQuant  255253  Lim et al."
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "prep_df_counts(gygi_counts, 'Lim et al.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Error Type</th>\n",
       "      <th>engine</th>\n",
       "      <th>values</th>\n",
       "      <th>DF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v1</td>\n",
       "      <td>276230</td>\n",
       "      <td>Lim et al.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v2 2.5%</td>\n",
       "      <td>158135</td>\n",
       "      <td>Lim et al.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v2 5%</td>\n",
       "      <td>180523</td>\n",
       "      <td>Lim et al.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>IonQuant 1%</td>\n",
       "      <td>364750</td>\n",
       "      <td>Lim et al.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>IonQuant 5%</td>\n",
       "      <td>413139</td>\n",
       "      <td>Lim et al.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>MaxQuant</td>\n",
       "      <td>255253</td>\n",
       "      <td>Lim et al.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Error Type            engine  values          DF\n",
       "0  PIP Identified Peak Traces       FlashLFQ v1  276230  Lim et al.\n",
       "1  PIP Identified Peak Traces  FlashLFQ v2 2.5%  158135  Lim et al.\n",
       "2  PIP Identified Peak Traces    FlashLFQ v2 5%  180523  Lim et al.\n",
       "3  PIP Identified Peak Traces       IonQuant 1%  364750  Lim et al.\n",
       "4  PIP Identified Peak Traces       IonQuant 5%  413139  Lim et al.\n",
       "5  PIP Identified Peak Traces          MaxQuant  255253  Lim et al."
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def prep_df_counts(df, name):\n",
    "    df = df.stack().reset_index()\n",
    "    df.columns = ['counts', 'engine', 'values']\n",
    "    df['DF'] = name\n",
    "    return df\n",
    "\n",
    "df1 = prep_df_counts(gygi_counts, 'Lim et al.')\n",
    "df1[\"values\"] = df1[\"values\"] / 20\n",
    "df2 = prep_df_counts(inhouse_counts, 'E. coli')\n",
    "df2[\"values\"] = df2[\"values\"] / 10\n",
    "df3 = prep_df_counts(kelly_counts, 'Single Cell')\n",
    "df3[\"values\"] = df3[\"values\"] / 7\n",
    "\n",
    "df = pd.concat([df1, df2, df3])\n",
    "\n",
    "df[\"Color\"] = df[\"engine\"]\n",
    "\n",
    "color = []\n",
    "for x in df[\"Color\"].tolist():\n",
    "    if (\"Foreign\" in x):\n",
    "        if('FlashLFQ v2 2.5%' in x):\n",
    "            color.append('FlashLFQ v2 2.5%')\n",
    "        elif('FlashLFQ v2 5%' in x):\n",
    "            color.append('FlashLFQ v2 5%')\n",
    "        elif(\"IonQuant 1%\" in x):\n",
    "            color.append('IonQuant 1%')\n",
    "        elif(\"IonQuant 5%\" in x):\n",
    "            color.append('IonQuant 5%')\n",
    "        elif('FlashLFQ v1' in x):\n",
    "            color.append('FlashLFQ v1')\n",
    "        else:\n",
    "            color.append(x[0:8])\n",
    "    else:\n",
    "        color.append(x)\n",
    "\n",
    "df[\"Software\"] = color"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>counts</th>\n",
       "      <th>engine</th>\n",
       "      <th>values</th>\n",
       "      <th>DF</th>\n",
       "      <th>Color</th>\n",
       "      <th>Software</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v1</td>\n",
       "      <td>13811.500000</td>\n",
       "      <td>Lim et al.</td>\n",
       "      <td>FlashLFQ v1</td>\n",
       "      <td>FlashLFQ v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v2 2.5%</td>\n",
       "      <td>7906.750000</td>\n",
       "      <td>Lim et al.</td>\n",
       "      <td>FlashLFQ v2 2.5%</td>\n",
       "      <td>FlashLFQ v2 2.5%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v2 5%</td>\n",
       "      <td>9026.150000</td>\n",
       "      <td>Lim et al.</td>\n",
       "      <td>FlashLFQ v2 5%</td>\n",
       "      <td>FlashLFQ v2 5%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>IonQuant 1%</td>\n",
       "      <td>18237.500000</td>\n",
       "      <td>Lim et al.</td>\n",
       "      <td>IonQuant 1%</td>\n",
       "      <td>IonQuant 1%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>IonQuant 5%</td>\n",
       "      <td>20656.950000</td>\n",
       "      <td>Lim et al.</td>\n",
       "      <td>IonQuant 5%</td>\n",
       "      <td>IonQuant 5%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>MaxQuant</td>\n",
       "      <td>12762.650000</td>\n",
       "      <td>Lim et al.</td>\n",
       "      <td>MaxQuant</td>\n",
       "      <td>MaxQuant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v1</td>\n",
       "      <td>18045.200000</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>FlashLFQ v1</td>\n",
       "      <td>FlashLFQ v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v2 2.5%</td>\n",
       "      <td>4716.100000</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>FlashLFQ v2 2.5%</td>\n",
       "      <td>FlashLFQ v2 2.5%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v2 5%</td>\n",
       "      <td>10621.700000</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>FlashLFQ v2 5%</td>\n",
       "      <td>FlashLFQ v2 5%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>IonQuant 1%</td>\n",
       "      <td>10719.300000</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>IonQuant 1%</td>\n",
       "      <td>IonQuant 1%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>IonQuant 5%</td>\n",
       "      <td>13098.700000</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>IonQuant 5%</td>\n",
       "      <td>IonQuant 5%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>MaxQuant</td>\n",
       "      <td>11881.400000</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>MaxQuant</td>\n",
       "      <td>MaxQuant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v1</td>\n",
       "      <td>11097.000000</td>\n",
       "      <td>Single Cell</td>\n",
       "      <td>FlashLFQ v1</td>\n",
       "      <td>FlashLFQ v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v2 2.5%</td>\n",
       "      <td>19.000000</td>\n",
       "      <td>Single Cell</td>\n",
       "      <td>FlashLFQ v2 2.5%</td>\n",
       "      <td>FlashLFQ v2 2.5%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>FlashLFQ v2 5%</td>\n",
       "      <td>2445.571429</td>\n",
       "      <td>Single Cell</td>\n",
       "      <td>FlashLFQ v2 5%</td>\n",
       "      <td>FlashLFQ v2 5%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>IonQuant 1%</td>\n",
       "      <td>468.000000</td>\n",
       "      <td>Single Cell</td>\n",
       "      <td>IonQuant 1%</td>\n",
       "      <td>IonQuant 1%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>IonQuant 5%</td>\n",
       "      <td>7458.428571</td>\n",
       "      <td>Single Cell</td>\n",
       "      <td>IonQuant 5%</td>\n",
       "      <td>IonQuant 5%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PIP Identified Peak Traces</td>\n",
       "      <td>MaxQuant</td>\n",
       "      <td>2390.714286</td>\n",
       "      <td>Single Cell</td>\n",
       "      <td>MaxQuant</td>\n",
       "      <td>MaxQuant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       counts            engine        values           DF  \\\n",
       "0  PIP Identified Peak Traces       FlashLFQ v1  13811.500000   Lim et al.   \n",
       "1  PIP Identified Peak Traces  FlashLFQ v2 2.5%   7906.750000   Lim et al.   \n",
       "2  PIP Identified Peak Traces    FlashLFQ v2 5%   9026.150000   Lim et al.   \n",
       "3  PIP Identified Peak Traces       IonQuant 1%  18237.500000   Lim et al.   \n",
       "4  PIP Identified Peak Traces       IonQuant 5%  20656.950000   Lim et al.   \n",
       "5  PIP Identified Peak Traces          MaxQuant  12762.650000   Lim et al.   \n",
       "0  PIP Identified Peak Traces       FlashLFQ v1  18045.200000      E. coli   \n",
       "1  PIP Identified Peak Traces  FlashLFQ v2 2.5%   4716.100000      E. coli   \n",
       "2  PIP Identified Peak Traces    FlashLFQ v2 5%  10621.700000      E. coli   \n",
       "3  PIP Identified Peak Traces       IonQuant 1%  10719.300000      E. coli   \n",
       "4  PIP Identified Peak Traces       IonQuant 5%  13098.700000      E. coli   \n",
       "5  PIP Identified Peak Traces          MaxQuant  11881.400000      E. coli   \n",
       "0  PIP Identified Peak Traces       FlashLFQ v1  11097.000000  Single Cell   \n",
       "1  PIP Identified Peak Traces  FlashLFQ v2 2.5%     19.000000  Single Cell   \n",
       "2  PIP Identified Peak Traces    FlashLFQ v2 5%   2445.571429  Single Cell   \n",
       "3  PIP Identified Peak Traces       IonQuant 1%    468.000000  Single Cell   \n",
       "4  PIP Identified Peak Traces       IonQuant 5%   7458.428571  Single Cell   \n",
       "5  PIP Identified Peak Traces          MaxQuant   2390.714286  Single Cell   \n",
       "\n",
       "              Color          Software  \n",
       "0       FlashLFQ v1       FlashLFQ v1  \n",
       "1  FlashLFQ v2 2.5%  FlashLFQ v2 2.5%  \n",
       "2    FlashLFQ v2 5%    FlashLFQ v2 5%  \n",
       "3       IonQuant 1%       IonQuant 1%  \n",
       "4       IonQuant 5%       IonQuant 5%  \n",
       "5          MaxQuant          MaxQuant  \n",
       "0       FlashLFQ v1       FlashLFQ v1  \n",
       "1  FlashLFQ v2 2.5%  FlashLFQ v2 2.5%  \n",
       "2    FlashLFQ v2 5%    FlashLFQ v2 5%  \n",
       "3       IonQuant 1%       IonQuant 1%  \n",
       "4       IonQuant 5%       IonQuant 5%  \n",
       "5          MaxQuant          MaxQuant  \n",
       "0       FlashLFQ v1       FlashLFQ v1  \n",
       "1  FlashLFQ v2 2.5%  FlashLFQ v2 2.5%  \n",
       "2    FlashLFQ v2 5%    FlashLFQ v2 5%  \n",
       "3       IonQuant 1%       IonQuant 1%  \n",
       "4       IonQuant 5%       IonQuant 5%  \n",
       "5          MaxQuant          MaxQuant  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<style>\n",
       "  #altair-viz-18b5b5208c1042618a88a6835bc30e32.vega-embed {\n",
       "    width: 100%;\n",
       "    display: flex;\n",
       "  }\n",
       "\n",
       "  #altair-viz-18b5b5208c1042618a88a6835bc30e32.vega-embed details,\n",
       "  #altair-viz-18b5b5208c1042618a88a6835bc30e32.vega-embed details summary {\n",
       "    position: relative;\n",
       "  }\n",
       "</style>\n",
       "<div id=\"altair-viz-18b5b5208c1042618a88a6835bc30e32\"></div>\n",
       "<script type=\"text/javascript\">\n",
       "  var VEGA_DEBUG = (typeof VEGA_DEBUG == \"undefined\") ? {} : VEGA_DEBUG;\n",
       "  (function(spec, embedOpt){\n",
       "    let outputDiv = document.currentScript.previousElementSibling;\n",
       "    if (outputDiv.id !== \"altair-viz-18b5b5208c1042618a88a6835bc30e32\") {\n",
       "      outputDiv = document.getElementById(\"altair-viz-18b5b5208c1042618a88a6835bc30e32\");\n",
       "    }\n",
       "    const paths = {\n",
       "      \"vega\": \"https://cdn.jsdelivr.net/npm/vega@5?noext\",\n",
       "      \"vega-lib\": \"https://cdn.jsdelivr.net/npm/vega-lib?noext\",\n",
       "      \"vega-lite\": \"https://cdn.jsdelivr.net/npm/vega-lite@5.16.3?noext\",\n",
       "      \"vega-embed\": \"https://cdn.jsdelivr.net/npm/vega-embed@6?noext\",\n",
       "    };\n",
       "\n",
       "    function maybeLoadScript(lib, version) {\n",
       "      var key = `${lib.replace(\"-\", \"\")}_version`;\n",
       "      return (VEGA_DEBUG[key] == version) ?\n",
       "        Promise.resolve(paths[lib]) :\n",
       "        new Promise(function(resolve, reject) {\n",
       "          var s = document.createElement('script');\n",
       "          document.getElementsByTagName(\"head\")[0].appendChild(s);\n",
       "          s.async = true;\n",
       "          s.onload = () => {\n",
       "            VEGA_DEBUG[key] = version;\n",
       "            return resolve(paths[lib]);\n",
       "          };\n",
       "          s.onerror = () => reject(`Error loading script: ${paths[lib]}`);\n",
       "          s.src = paths[lib];\n",
       "        });\n",
       "    }\n",
       "\n",
       "    function showError(err) {\n",
       "      outputDiv.innerHTML = `<div class=\"error\" style=\"color:red;\">${err}</div>`;\n",
       "      throw err;\n",
       "    }\n",
       "\n",
       "    function displayChart(vegaEmbed) {\n",
       "      vegaEmbed(outputDiv, spec, embedOpt)\n",
       "        .catch(err => showError(`Javascript Error: ${err.message}<br>This usually means there's a typo in your chart specification. See the javascript console for the full traceback.`));\n",
       "    }\n",
       "\n",
       "    if(typeof define === \"function\" && define.amd) {\n",
       "      requirejs.config({paths});\n",
       "      require([\"vega-embed\"], displayChart, err => showError(`Error loading script: ${err.message}`));\n",
       "    } else {\n",
       "      maybeLoadScript(\"vega\", \"5\")\n",
       "        .then(() => maybeLoadScript(\"vega-lite\", \"5.16.3\"))\n",
       "        .then(() => maybeLoadScript(\"vega-embed\", \"6\"))\n",
       "        .catch(showError)\n",
       "        .then(() => displayChart(vegaEmbed));\n",
       "    }\n",
       "  })({\"config\": {\"view\": {\"continuousWidth\": 300, \"continuousHeight\": 300, \"strokeOpacity\": 0}, \"axis\": {\"labelFontSize\": 18, \"titleFontSize\": 20}, \"legend\": {\"labelFontSize\": 18, \"titleFontSize\": 22}}, \"data\": {\"name\": \"data-4125ffe7496c0baa05a814e9a9cdc4f0\"}, \"mark\": {\"type\": \"bar\"}, \"encoding\": {\"color\": {\"field\": \"Software\", \"legend\": {\"values\": [\"FlashLFQ v1\", \"FlashLFQ v2 2.5%\", \"FlashLFQ v2 5%\", \"IonQuant 1%\", \"IonQuant 5%\", \"MaxQuant\"]}, \"scale\": {\"range\": [\"#13459c\", \"#118C53\", \"#118c11\", \"#E54501\", \"#9f040e\", \"#44485D\"]}, \"sort\": [\"FlashLFQ v1\", \"FlashLFQ v2 2.5%\", \"FlashLFQ v2 5%\", \"IonQuant 1%\", \"IonQuant 5%\", \"MaxQuant\"], \"type\": \"nominal\"}, \"column\": {\"field\": \"DF\", \"header\": {\"labelFontSize\": 24}, \"spacing\": 45, \"title\": null, \"type\": \"nominal\"}, \"order\": {\"field\": \"Software\", \"sort\": \"ascending\", \"type\": \"nominal\"}, \"x\": {\"axis\": {\"labelAngle\": -45}, \"field\": \"engine\", \"sort\": [\"FlashLFQ v1\", \"FlashLFQ v2 2.5%\", \"FlashLFQ v2 5%\", \"IonQuant 1%\", \"IonQuant 5%\", \"MaxQuant\"], \"title\": null, \"type\": \"nominal\"}, \"y\": {\"aggregate\": \"sum\", \"axis\": {\"grid\": false, \"title\": \"PIP Identifications Per File\"}, \"field\": \"values\", \"type\": \"quantitative\"}}, \"height\": 350, \"width\": 200, \"$schema\": \"https://vega.github.io/schema/vega-lite/v5.16.3.json\", \"datasets\": {\"data-4125ffe7496c0baa05a814e9a9cdc4f0\": [{\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"FlashLFQ v1\", \"values\": 13811.5, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v1\", \"Software\": \"FlashLFQ v1\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"FlashLFQ v2 2.5%\", \"values\": 7906.75, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v2 2.5%\", \"Software\": \"FlashLFQ v2 2.5%\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 9026.15, \"DF\": \"Lim et al.\", \"Color\": \"FlashLFQ v2 5%\", \"Software\": \"FlashLFQ v2 5%\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"IonQuant 1%\", \"values\": 18237.5, \"DF\": \"Lim et al.\", \"Color\": \"IonQuant 1%\", \"Software\": \"IonQuant 1%\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"IonQuant 5%\", \"values\": 20656.95, \"DF\": \"Lim et al.\", \"Color\": \"IonQuant 5%\", \"Software\": \"IonQuant 5%\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"MaxQuant\", \"values\": 12762.65, \"DF\": \"Lim et al.\", \"Color\": \"MaxQuant\", \"Software\": \"MaxQuant\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"FlashLFQ v1\", \"values\": 18045.2, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v1\", \"Software\": \"FlashLFQ v1\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"FlashLFQ v2 2.5%\", \"values\": 4716.1, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v2 2.5%\", \"Software\": \"FlashLFQ v2 2.5%\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 10621.7, \"DF\": \"E. coli\", \"Color\": \"FlashLFQ v2 5%\", \"Software\": \"FlashLFQ v2 5%\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"IonQuant 1%\", \"values\": 10719.3, \"DF\": \"E. coli\", \"Color\": \"IonQuant 1%\", \"Software\": \"IonQuant 1%\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"IonQuant 5%\", \"values\": 13098.7, \"DF\": \"E. coli\", \"Color\": \"IonQuant 5%\", \"Software\": \"IonQuant 5%\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"MaxQuant\", \"values\": 11881.4, \"DF\": \"E. coli\", \"Color\": \"MaxQuant\", \"Software\": \"MaxQuant\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"FlashLFQ v1\", \"values\": 11097.0, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v1\", \"Software\": \"FlashLFQ v1\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"FlashLFQ v2 2.5%\", \"values\": 19.0, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v2 2.5%\", \"Software\": \"FlashLFQ v2 2.5%\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"FlashLFQ v2 5%\", \"values\": 2445.5714285714284, \"DF\": \"Single Cell\", \"Color\": \"FlashLFQ v2 5%\", \"Software\": \"FlashLFQ v2 5%\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"IonQuant 1%\", \"values\": 468.0, \"DF\": \"Single Cell\", \"Color\": \"IonQuant 1%\", \"Software\": \"IonQuant 1%\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"IonQuant 5%\", \"values\": 7458.428571428572, \"DF\": \"Single Cell\", \"Color\": \"IonQuant 5%\", \"Software\": \"IonQuant 5%\"}, {\"counts\": \"PIP Identified Peak Traces\", \"engine\": \"MaxQuant\", \"values\": 2390.714285714286, \"DF\": \"Single Cell\", \"Color\": \"MaxQuant\", \"Software\": \"MaxQuant\"}]}}, {\"mode\": \"vega-lite\"});\n",
       "</script>"
      ],
      "text/plain": [
       "alt.Chart(...)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "label_order = [ \n",
    " 'FlashLFQ v1', \n",
    " 'FlashLFQ v2 2.5%', \n",
    " 'FlashLFQ v2 5%', \n",
    "  'IonQuant 1%', \n",
    "  'IonQuant 5%', \n",
    "  'MaxQuant'\n",
    "  ]\n",
    "\n",
    "alt.Chart(df).mark_bar().encode(\n",
    "\n",
    "    # tell Altair which field to group columns on\n",
    "    x=alt.X('engine:N', title=None, sort = [\"FlashLFQ v1\", 'FlashLFQ v2 2.5%', 'FlashLFQ v2 5%', \"IonQuant 1%\", \"IonQuant 5%\", \"MaxQuant\"], axis=alt.Axis(labelAngle=-45)),\n",
    "\n",
    "    # tell Altair which field to use as Y values and how to calculate\n",
    "    y=alt.Y('sum(values):Q', \n",
    "        axis=alt.Axis(\n",
    "            grid=False,\n",
    "            title=\"PIP Identifications Per File\")),\n",
    "\n",
    "    # tell Altair which field to use to use as the set of columns to be  represented in each group\n",
    "    column=alt.Column('DF:N', title=None, header=alt.Header(labelFontSize=24), spacing=45),\n",
    "\n",
    "    # tell Altair which field to use for color segmentation \n",
    "    color=alt.Color('Software:N', sort = label_order,\n",
    "            scale=alt.Scale(\n",
    "                # make it look pretty with an enjoyable color pallet\n",
    "                range=[ '#13459c',\n",
    "                       '#118C53',\n",
    "                       '#118c11',\n",
    "                       '#E54501',\n",
    "                       '#9f040e',\n",
    "                       '#44485D'\n",
    "                       ],\n",
    "            ),\n",
    "            legend=alt.Legend(values=[\"FlashLFQ v1\", \"FlashLFQ v2 2.5%\", \"FlashLFQ v2 5%\", \"IonQuant 1%\", \"IonQuant 5%\", \"MaxQuant\"])\n",
    "        ),\n",
    "     \n",
    "    order = alt.Order(\"Software:N\", sort = 'ascending')\n",
    "        )\\\n",
    "    .configure_view(\n",
    "        # remove grid lines around column clusters\n",
    "        strokeOpacity=0   \n",
    "    )\\\n",
    "        .properties(width = 200, height = 350)\\\n",
    "        .configure_legend(titleFontSize = 22, labelFontSize =18)\\\n",
    "        .configure_axis(titleFontSize = 20, labelFontSize =18)"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
